---
title: "Analysis of ZFX publicly available ChIP-seq data"
output: html_notebook
---

#Setup
```{r}
myPath <- #ADD PATH TO GITHUB FOLDER
# library("ggplot2")
# library("ggrepel")
# library("reshape2")
# library("deming")

data_summary <- function(x){
  m <- median(x, na.rm=TRUE)
  ymin <- quantile(x, na.rm=TRUE)[2]
  ymax <- quantile(x, na.rm=TRUE)[4]
  test <- c("y"=m,"ymin"=ymin,"ymax"=ymax)
  names(test) <- c("y","ymin","ymax")
  return(test)
}
```

# Bring in annotations
```{r}
geneAnno <- na.omit(read.delim(file=paste0(myPath,"Annotations/geneAnno_proteinCoding_lncRNA_v107_20230426.txt"), stringsAsFactors = FALSE))
#Main "Gene" name will be ensembl84
colnames(geneAnno)[c(2,4:6)] <- c("Gene","chr","start","stop")
#Only include genes in both v84 and v107
geneAnno <- geneAnno[! is.na(geneAnno$Gene),]
geneAnno <- geneAnno[! duplicated(geneAnno$gene_name.107) & ! geneAnno$gene_name.107 %in% c("") & ! duplicated(geneAnno$Gene),]
geneAnno_x <- geneAnno[geneAnno$chr == "chrX",]
geneAnno_x <- na.omit(geneAnno_x)
geneAnno_y <- geneAnno[geneAnno$chr == "chrY",]
X_genes_all <-as.character(geneAnno_x$Gene)
Y_genes_all <-as.character(geneAnno_y$Gene)
sexChrom_genes <- c(X_genes_all,Y_genes_all)
PAR1_genes <- geneAnno_x[geneAnno_x$start < 2691188,]$Gene
PAR2_genes <- c("SPRY3","VAMP7","IL9R","WASH6P")
PAR_genes_all <- c(PAR1_genes,PAR2_genes)
NPX_genes <- X_genes_all[! X_genes_all %in% PAR_genes_all]
NPY_genes <- Y_genes_all[! Y_genes_all %in% PAR_genes_all]


#Colors for figures
myOrange <- "#d95f0290"
myOrange_100 <- "#d95f02"
myOrange_80 <- "#d95f0280"
myPurple <- "#7570B390"
myGreen_95 <- "#1B9E7795"
chr21Col <- "#DA70D690"
```

#Get tss annotations
```{r, eval=FALSE}
myString <- function (x){
  strsplit(x, "[.]")[[1]][1]
}

gencode_ucsc <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/gencodev24_comp.bed"),stringsAsFactors = FALSE, header = TRUE)
gencode_ucsc$TSS_start <- 0
gencode_ucsc$TSS_end <- 0
gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "+", "TSS_start"] <- gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "+", "hg38.wgEncodeGencodeCompV24.txStart"]
gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "+", "TSS_end"] <- gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "+", "hg38.wgEncodeGencodeCompV24.txStart"] + 1

gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "-", "TSS_end"] <- gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "-", "hg38.wgEncodeGencodeCompV24.txEnd"]
gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "-", "TSS_start"] <- gencode_ucsc[gencode_ucsc$hg38.wgEncodeGencodeCompV24.strand == "-", "hg38.wgEncodeGencodeCompV24.txEnd"] - 1

gencode_v24_tss <- gencode_ucsc[,c(2,9,10,3,6,7,8)]
colnames(gencode_v24_tss) <- c("chr","start","stop","strand","gene_ID","gene_name","transcript_ID")

gencode_v24_tss$gene_ID <- sapply(X=gencode_v24_tss$gene_ID, FUN=myString)
gencode_v24_tss$transcript_ID <- sapply(X=gencode_v24_tss$transcript_ID, FUN=myString)

#remove redundant TSSs
gencode_v24_tss$TSS_name <- paste0(gencode_v24_tss$chr, "_",gencode_v24_tss$start)

gencode_v24_tss_unique <- gencode_v24_tss[! duplicated(gencode_v24_tss$TSS_name),]
rownames(gencode_v24_tss_unique) <- NULL
gencode_v24_tss_unique[65954,c("start")] <- format(x=gencode_v24_tss_unique[65954,c("start")], scientific = FALSE)
gencode_v24_tss_unique[65954,c("stop")] <- format(x=gencode_v24_tss_unique[65954,c("stop")], scientific = FALSE)

write.table(x=gencode_v24_tss_unique, file="gencode_v24_tss_20220608_allChr.bed", col.names = FALSE, row.names = FALSE, sep="\t", quote=FALSE)
```

#Get closest gene to each peak
```{bash, eval=FALSE}

#DOWNLOAD THE LISTED FILES FROM ENCODE 

#Sort TSS file
bedtools sort -i gencode_v24_tss_20220608_allChr.bed > gencode_v24_tss_20220608_allChr_sorted.bed

#Encode 
#HCT116
bedtools sort -i ENCODE/ENCFF858YWR.bed > HCT116_ZFX_sorted.bed

bedtools closest -d -t first -a HCT116_ZFX_sorted.bed -b gencode_v24_tss_20220608_allChr_sorted.bed > HCT116_ZFX_closestTSS.bed

#C42B
bedtools sort -i ENCODE/ENCFF160AXQ.bed > C42B_ZFX_sorted.bed

bedtools closest -d -t first -a C42B_ZFX_sorted.bed -b gencode_v24_tss_20220608_allChr_sorted.bed > C42B_ZFX_closestTSS.bed

#MCF7
bedtools sort -i ENCODE/ENCFF861DOL.bed > mcf7_ZFX_sorted.bed

bedtools closest -d -t first -a mcf7_ZFX_sorted.bed -b gencode_v24_tss_20220608_allChr_sorted.bed > MCF7_ZFX_closestTSS.bed

#HEK-293T
bedtools sort -i ENCODE/ENCFF292HYE.bed > hek293t_ZFX_sorted.bed

bedtools closest -d -t first -a hek293t_ZFX_sorted.bed -b gencode_v24_tss_20220608_allChr_sorted.bed > HEK293T_ZFX_closestTSS.bed
```


#Start here with processed data
#Bring in all of the genes with binding peaks in each dataset and compile into one table
```{r}
library(data.table)
myCol_names <- c("chr_peak","start_peak","stop_peak","name","score","strand","signalValue","pValue","qValue","peak", "chr","start","stop","strand","gene_id","gene","transcript_id","tss_name","distance")

#HCT116 #84 annotations!
hct116_EN_closest <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/HCT116_ZFX_closestTSS.bed"), header=FALSE, stringsAsFactors = FALSE)
colnames(hct116_EN_closest) <- myCol_names
hct116_EN_closest_protein <- hct116_EN_closest[hct116_EN_closest$gene %in% geneAnno$Gene,]
#restrict distance to 1kb
hct116_EN_closest_1kb <- hct116_EN_closest_protein[hct116_EN_closest_protein$distance <= 1000,]
hct116_EN_closest_1kb <- data.table(hct116_EN_closest_1kb)
hct116_peak_signals <- hct116_EN_closest_1kb[,.(hct116_signalValue = max(signalValue)), by="gene"]

#C42B
c42b_EN_closest <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/C42B_ZFX_closestTSS.bed"), header=FALSE, stringsAsFactors = FALSE)
colnames(c42b_EN_closest) <- myCol_names
c42b_EN_closest_protein <- c42b_EN_closest[c42b_EN_closest$gene %in% geneAnno$Gene,]
#restrict distance to 1kb
c42b_EN_closest_1kb <- c42b_EN_closest_protein[c42b_EN_closest_protein$distance <= 1000,]
c42b_EN_closest_1kb <- data.table(c42b_EN_closest_1kb)
c42b_peak_signals <- c42b_EN_closest_1kb[,.(c42b_signalValue = max(signalValue)), by="gene"]


#MCF7
mcf7_EN_closest <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/MCF7_ZFX_closestTSS.bed"), header=FALSE, stringsAsFactors = FALSE)
colnames(mcf7_EN_closest) <- myCol_names
mcf7_EN_closest_protein <- mcf7_EN_closest[mcf7_EN_closest$gene %in% geneAnno$Gene,]
#restrict distance to 1kb
mcf7_EN_closest_1kb <- mcf7_EN_closest_protein[mcf7_EN_closest_protein$distance <= 1000,]
mcf7_EN_closest_1kb <- data.table(mcf7_EN_closest_1kb)
mcf7_peak_signals <- mcf7_EN_closest_1kb[,.(mcf7_signalValue = max(signalValue)), by="gene"]


#HEK-293T
hek293t_EN_closest <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/HEK293T_ZFX_closestTSS.bed"), header=FALSE, stringsAsFactors = FALSE)
colnames(hek293t_EN_closest) <- myCol_names
hek293t_EN_closest_protein <- hek293t_EN_closest[hek293t_EN_closest$gene %in% geneAnno$Gene,]
#restrict distance to 1kb
hek293t_EN_closest_1kb <- hek293t_EN_closest_protein[hek293t_EN_closest_protein$distance <= 1000,]
hek293t_EN_closest_1kb <- data.table(hek293t_EN_closest_1kb)
hek293t_peak_signals <- hek293t_EN_closest_1kb[,.(hek293t_signalValue = max(signalValue)), by="gene"]

#Bring together all of the gene annotations into one file with one column per cell type and 0 or 1 for whether the gene is bound.
geneAnno <- geneAnno[!is.na(geneAnno$Gene),]
binary_gene_bound <- data.frame(row.names = geneAnno$Gene)
binary_gene_bound$hek293t <- as.numeric((rownames(binary_gene_bound) %in% hek293t_EN_closest_1kb$gene))
binary_gene_bound$hct116 <- as.numeric((rownames(binary_gene_bound) %in% hct116_EN_closest_1kb$gene) )
binary_gene_bound$mcf7 <- as.numeric((rownames(binary_gene_bound) %in% mcf7_EN_closest_1kb$gene) )
binary_gene_bound$c42b <- as.numeric((rownames(binary_gene_bound) %in% c42b_EN_closest_1kb$gene))

binary_gene_bound$n_peaks <- apply(X = binary_gene_bound, MARGIN = 1, FUN = sum)
write.table(x=binary_gene_bound, file="ZFX_bound_within_1kb_TSS_allDatasets_allChr.txt", row.names=TRUE, col.names=TRUE, sep="\t", quote=FALSE)


#Put together the peak levels for each gene
merge_peaks_1 <- merge(x=hct116_peak_signals, y=c42b_peak_signals, by="gene", all=TRUE)
merge_peaks_2 <- merge(x=merge_peaks_1, y=mcf7_peak_signals, by="gene", all=TRUE)
all_zfx_peaks_promoter <- merge(x=merge_peaks_2, y=hek293t_peak_signals, by="gene", all=TRUE)

apply(X=all_zfx_peaks_promoter[,c(2:5)], MARGIN = 2, FUN = function(x){ median(x, na.rm=TRUE)})
write.table(x = all_zfx_peaks_promoter, file="all_zfx_peaks_promoter_signal_allChr.txt", quote=FALSE, sep = "\t", col.names = TRUE, row.names=FALSE)
```

#Bring in expression data so we can figure out for each cell type if the gene is expressed [and so don't count a non-expressed as negative data.]
```{r}
library(stringr)
myString <- function (x){
  strsplit(x, "[.]")[[1]][1]
}

myMedian <- function(x){
  median(x, na.rm = TRUE)
}

#fib expressed data
load(file=paste0(myPath,"Linear_regressions/Fibroblasts/fib_expressed.rda"))
fib_anno_exp <- fib_expressed$expressedGenes 
fib_anno_exp_encode <- geneAnno[geneAnno$Gene %in% fib_anno_exp, "gene_id.84"]

#Encode data - DOWNLOAD FROM ENCODE TO RUN!
#HCT116
hct116_expn <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/ENCODE/ENCFF435PHM.tsv"), header=TRUE, stringsAsFactors = FALSE)
hct116_expn$gene_id <- sapply(X = hct116_expn$gene_id, FUN = myString)
hct116_expn_anno <- merge(x= hct116_expn, by.x = "gene_id", y=geneAnno, by.y="gene_id.84", all.y=TRUE) #no issue with merge to gene_id.84

HCT116_expression <- hct116_expn_anno[! str_detect(string = hct116_expn_anno$transcript_id.s., pattern = "PAR_Y"),]
HCT116_expression <- HCT116_expression[! duplicated(HCT116_expression$gene_id) & !is.na(HCT116_expression$gene_id),]
rownames(HCT116_expression) <- HCT116_expression$gene_id
HCT116_expressed <- HCT116_expression[HCT116_expression$TPM >= 1,]

# write.table(x=HCT116_expressed, file = "HCT116_expressed.txt", quote=FALSE, sep="\t", col.names=TRUE, row.names=FALSE)


#MCF7
mcf7_expn <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/ENCODE/ENCFF721BRA.tsv"), header=TRUE, stringsAsFactors = FALSE)
mcf7_expn$gene_id <- sapply(X = mcf7_expn$gene_id, FUN = myString)
mcf7_expn_anno <- merge(x= mcf7_expn, by.x = "gene_id", y=geneAnno, by.y="gene_id.84", all.y=TRUE)
mcf7_expression <- mcf7_expn_anno[! str_detect(string = mcf7_expn_anno$transcript_id.s., pattern = "PAR_Y"),]
colnames(mcf7_expression)[6] <- "TPM_encode"


#MCF7 --> From Rhie 2018
Rhie_2018_metadata <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/Rhie_2018/rhie_c42b_mcf7_metadata.txt"), stringsAsFactors = FALSE)
Rhie_2018_metadata_mcf7 <- Rhie_2018_metadata[Rhie_2018_metadata$Cell_Line == "MCF7",]

# library(tximport)
# annofile <- read.delim(file=paste0(myPath,"Annotations/annotation_with_ERCC.txt"), sep=" ")
# tx2gene <- data.frame("TXNAME"=annofile$transcript_id, "GENEID"=annofile$gene_name)

#Do tximport 
# txi_mcf7_files <- paste0(#PATH TO Rhie et al 2018 files pseudoaligned with Kallisto)
# names(txi_mcf7_files) <- Rhie_2018_metadata_mcf7$Run
# txi_mcf7 <- tximport(txi_mcf7_files, type = "kallisto", tx2gene = tx2gene)
# save(txi_mcf7,file="txi_mcf7.rda")
load(file="txi_mcf7.rda")
mcf7_tpm_Rhie <- data.frame(txi_mcf7$abundance)
mcf7_tpm_Rhie_anno <- merge(x= mcf7_tpm_Rhie, by.x = 0, y=geneAnno, by.y="Gene", all.y=TRUE)

#Merge Encode and Rhie
mcf7_merged <- merge(x=mcf7_expression[,c(1,6)], by.x="gene_id", y=mcf7_tpm_Rhie_anno[,c(1:5,11)], by.y="gene_id.84", all.x=TRUE)
mcf7_merged <- mcf7_merged[! duplicated(mcf7_merged$Row.names) & !is.na(mcf7_merged$Row.names),]
rownames(mcf7_merged) <- mcf7_merged$Row.names
mcf7_merged <- mcf7_merged[,c(-3)]
mcf7_merged$median_TPM <-as.numeric(apply(X = mcf7_merged[,c(2:5)], MARGIN = 1, FUN = myMedian))
rownames(mcf7_merged) <- mcf7_merged$gene_id

mcf7_expressed <- mcf7_merged[mcf7_merged$median_TPM >= 1,]
# write.table(x=mcf7_expressed, file = "mcf7_expressed.txt", quote=FALSE, sep="\t", col.names=TRUE, row.names=FALSE)


#C42B --> From Rhie 2018
Rhie_2018_metadata_c42b <- Rhie_2018_metadata[Rhie_2018_metadata$Cell_Line == "C42B",]

#Do tximport for c42b
# txi_c42b_files <- paste0(#PATH TO Rhie et al 2018 files pseudoaligned with Kallisto)
# names(txi_c42b_files) <- Rhie_2018_metadata_c42b$Run
# txi_c42b <- tximport(txi_c42b_files, type = "kallisto", tx2gene = tx2gene)
# save(txi_c42b,file="txi_c42b.rda")
load(file="txi_c42b.rda")

c42b_tpm_Rhie <- data.frame(txi_c42b$abundance)
c42b_tpm_Rhie$median_TPM <- as.numeric(apply(X = c42b_tpm_Rhie, MARGIN = 1, FUN = myMedian))
c42b_tpm_Rhie <- c42b_tpm_Rhie[geneAnno$Gene,]
c42b_tpm_Rhie_anno <- merge(x=c42b_tpm_Rhie, by.x = 0, y=geneAnno, by.y="Gene", all.y=TRUE)
c42b_tpm_Rhie_anno <- c42b_tpm_Rhie_anno[!duplicated(c42b_tpm_Rhie_anno$gene_id.84) & ! is.na(c42b_tpm_Rhie_anno$gene_id.84),]
rownames(c42b_tpm_Rhie_anno) <- c42b_tpm_Rhie_anno$gene_id.84
c42b_expressed <- c42b_tpm_Rhie_anno[c42b_tpm_Rhie_anno$median_TPM >= 1,]
# write.table(x=c42b_expressed, file = "c42b_expressed.txt", quote=FALSE, sep="\t", col.names=TRUE, row.names=FALSE)


#HEK-293T --> Get from Ni 2020
#read in the metadata table and restrict to the appropriate samples
metadata_Ni <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/Ni_2020/Ni_metadata.txt"))
metadata_Ni <- metadata_Ni[metadata_Ni$Condition == "Control",]

#Ni files -->
# Ni_files <- paste0(#PATH TO Ni et al files pseudoaligned with Kallisto)
# names(Ni_files) <- metadata_Ni$BioSample
# txi_Ni <- tximport(Ni_files, type = "kallisto", tx2gene = tx2gene)
# save(txi_Ni, file = "txi_Ni.rda")
load(file = "txi_Ni.rda")
hek293t_tpm_Ni <- data.frame(txi_Ni$abundance)
hek293t_tpm_Ni$median_TPM <- apply(X = hek293t_tpm_Ni, MARGIN = 1, FUN = myMedian)
hek293t_tpm_Ni <- hek293t_tpm_Ni[geneAnno$Gene,]
hek293t_tpm_Ni_anno <- merge(x=hek293t_tpm_Ni, by.x = 0, y=geneAnno, by.y="Gene", all.y=TRUE)
hek293t_tpm_Ni_anno <- hek293t_tpm_Ni_anno[! duplicated(hek293t_tpm_Ni_anno$gene_id.84) & !is.na(hek293t_tpm_Ni_anno$gene_id.84),]
row.names(hek293t_tpm_Ni_anno) <- hek293t_tpm_Ni_anno$gene_id.84

hek293t_expressed <- hek293t_tpm_Ni_anno[hek293t_tpm_Ni_anno$median_TPM >= 1,]
# write.table(x=hek293t_expressed, file = "hek293t_expressed.txt", quote=FALSE, sep="\t", col.names=TRUE, row.names=FALSE)

#Bring together all of the gene expression into one file with one column per cell type and 0 or 1 for whether the gene is expressed --> The annotation is definitely v24 for the 293T and C4-2b
geneAnno <- geneAnno[! duplicated(geneAnno$gene_id.84),]
binary_gene_exp <- data.frame(row.names = geneAnno$gene_id.84)
binary_gene_exp$Gene <- geneAnno$Gene
binary_gene_exp$fib <- as.numeric(rownames(binary_gene_exp) %in% fib_anno_exp_encode)
binary_gene_exp$hek293t <- as.numeric(rownames(binary_gene_exp) %in% hek293t_expressed$gene_id.84) 

binary_gene_exp$hct116 <- as.numeric(rownames(binary_gene_exp) %in% HCT116_expressed$gene_id)

binary_gene_exp$mcf7 <- as.numeric(rownames(binary_gene_exp) %in% mcf7_expressed$gene_id)

binary_gene_exp$c42b <- as.numeric(rownames(binary_gene_exp) %in% c42b_expressed$gene_id.84)

binary_gene_exp$n_exp <- apply(X = binary_gene_exp[,c(2:6)], MARGIN = 1, FUN = sum)
# write.table(x=binary_gene_exp, file="binary_gene_expn_allDatasets_allChr.txt", row.names=TRUE, col.names=TRUE, sep="\t", quote=FALSE)
```

#Merge ChIP-seq and expn
```{r}
#Need to make sure that they both use the same rownames. 
binary_gene_bound_mod <- binary_gene_bound[,c(1:4)]

binary_gene_exp <- binary_gene_exp[! duplicated(binary_gene_exp$Gene) & ! is.na(binary_gene_exp$Gene),]
rownames(binary_gene_exp) <- binary_gene_exp$Gene

binary_gene_bound_mod[binary_gene_exp$hek293t == FALSE,"hek293t"] <- NA
binary_gene_bound_mod[binary_gene_exp$hct116 == FALSE,"hct116"] <- NA
binary_gene_bound_mod[binary_gene_exp$mcf7 == FALSE,"mcf7"] <- NA
binary_gene_bound_mod[binary_gene_exp$c42b == FALSE,"c42b"] <- NA

myMean <- function(x){
  mean(x, na.rm=TRUE)
}
binary_gene_bound_mod$pct_peaks <- apply(X = binary_gene_bound_mod, MARGIN = 1, FUN = myMean)
binary_gene_bound_mod[binary_gene_bound_mod$pct_peaks =="NaN","pct_peaks"] <- NA

write.table(x=binary_gene_bound_mod, file="binary_ZFX_binding_withExpn.txt", quote=FALSE, sep="\t", col.names=TRUE, row.names = TRUE)
```


