---
title: "CRISPRi_ZFX_ZFY"
output: html_notebook
---

#Setup
```{r}
myPath <- #ADD PATH TO GITHUB FOLDER
#load DESeq2 library
suppressPackageStartupMessages(library("DESeq2"))
library(ggrepel)

#Bring in functions!
source(file=paste0(myPath,"Scripts/Function_ASR-QQmanhattan_20221103.R"))
source(file=paste0(myPath,"Scripts/Function_DESeqManhattan_noM_col_20221108.R"))
source(file=paste0(myPath,"Scripts/Function_twoGroupVenn_geneNames_20221103.R"))
source(file=paste0(myPath,"Scripts/Function_threeGroupVenn_geneNames_20221202.R"))
source(file=paste0(myPath,"Scripts/Function_makeSymmetric_X_Y_20221103.R"))
source(file=paste0(myPath,"Scripts/Function_paper2_corrPlot_deming_20230419.R"))
source(file=paste0(myPath,"Scripts/Function_deseqFactorAnalysis_20221103.R"))



#Set up sex chromosome gene lists, using gencode v84 as "Gene" list
geneAnno <- na.omit(read.delim(file=paste0(myPath,"Annotations/geneAnno_proteinCoding_lncRNA_v107_20230426.txt"), stringsAsFactors = FALSE))
#Main "Gene" name will be ensembl84
colnames(geneAnno)[c(2,4:6)] <- c("Gene","chr","start","stop")
#Only include genes in both v84 and v107
geneAnno <- geneAnno[! is.na(geneAnno$Gene),]
geneAnno <- geneAnno[! duplicated(geneAnno$gene_name.107) & ! geneAnno$gene_name.107 %in% c("") & ! duplicated(geneAnno$Gene),]
geneAnno_x <- geneAnno[geneAnno$chr == "chrX",]
geneAnno_y <- geneAnno[geneAnno$chr == "chrY",]
X_genes_all <-as.character(geneAnno_x$Gene)
Y_genes_all <-as.character(geneAnno_y$Gene)
sexChrom_genes <- c(X_genes_all,Y_genes_all)
PAR1_genes <- geneAnno_x[geneAnno_x$start < 2691188,]$Gene
PAR2_genes <- c("SPRY3","VAMP7","IL9R","WASH6P")
PAR_genes_all <- c(PAR1_genes,PAR2_genes)
NPX_genes <- X_genes_all[! X_genes_all %in% PAR_genes_all]
NPY_genes <- Y_genes_all[! Y_genes_all %in% PAR_genes_all]
autosome_genes <- geneAnno[! geneAnno$chr %in% c("chrX","chrY"),"Gene"]

data_summary <- function(x){
  m <- median(x, na.rm=TRUE)
  ymin <- quantile(x, na.rm=TRUE)[2]
  ymax <- quantile(x, na.rm=TRUE)[4]
  test <- c("y"=m,"ymin"=ymin,"ymax"=ymax)
  names(test) <- c("y","ymin","ymax")
  return(test)
}
```

#Bring in metadata
```{r}
#read in the metadata table and restrict to the appropriate samples
#Make sure the categorical variables are treated as such by adding characters.
metadata <- read.delim("ZFX_ZFY_CRISPRi_metadata.txt", stringsAsFactors = FALSE)
# metadata$kallisto_path <- paste0(#PATHS TO FILES#)
metadata$batch_libprep <- paste0("BL",metadata$batch_libprep)
metadata$cell_line <- paste0("WHT",metadata$cell_line)

#separate analysis for double kds because these were run in a different experiment 
metadata_dkd <- metadata[metadata$genotype %in% c("Control", "ZFX_ZFY_kd") & metadata$batch_libprep %in% c("BL20210106","BL20210108") & metadata$cell_line %in% c("WHT7496", "WHT7497", "WHT7506"),] 

#Do separate analysis for single kds 
metadata_skd <- metadata[! metadata$sample %in% metadata_dkd$sample | (metadata$sample %in% c("7497_IG1_20201218","7497_IG3_20201218")),] 
#separate analyses for male and female single kds
metadata_ZFX_Y_F_skd <-  metadata_skd[metadata_skd$karyotype == "XX",]
metadata_ZFX_Y_M_skd <-  metadata_skd[metadata_skd$karyotype == "XY",]
```

#Bring in ZFX ChIP-seq data
```{r}
zfx_binding <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/binary_ZFX_binding_withExpn.txt"), stringsAsFactors = FALSE)

mySum <- function(x){
  sum(x,na.rm=TRUE)
}

zfx_binding$sum_peaks <- apply(X = zfx_binding[,c(1:4)], MARGIN = 1, FUN = mySum)
zfx_binding[is.na(zfx_binding$pct_peaks), "sum_peaks"] <- NA

zfx_binding_signal <- read.delim(file=paste0(myPath,"ZFX_ChIPseq/all_zfx_peaks_promoter_signal_allChr.txt"), stringsAsFactors = FALSE)
```

#Bring in X and Y response
```{r}
lcl_fib_results <- read.delim(file=paste0(myPath,"Linear_regressions/lcl_fib_XandY_results_auto_expressed.txt"), stringsAsFactors = FALSE)
rownames(lcl_fib_results) <- lcl_fib_results$Gene.y  #gene 84
colnames(lcl_fib_results)[colnames(lcl_fib_results) == "Gene.y"] <- "gene_name.84"
fib_results <- lcl_fib_results[! is.na(lcl_fib_results$fib_padj_X),]
fib_results$fib_log2FC_XY_avg <- (fib_results$fib_log2FC_X + fib_results$fib_log2FC_Y)/2
fib_results$fib_lfcSE_XY_avg <- (fib_results$fib_lfcSE_X + fib_results$fib_lfcSE_Y)/2
fib_results_sig_x <- fib_results[fib_results$fib_padj_X < 0.05,]
fib_results_sig_y <- fib_results[fib_results$fib_padj_Y < 0.05,]
fib_results_sig_x_y <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_padj_Y < 0.05,]
fib_results_sig_x_only <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_padj_Y >= 0.05,]
fib_results_sig_y_only <- fib_results[fib_results$fib_padj_Y < 0.05 & fib_results$fib_padj_X >= 0.05,]

#compare with ZFX ChIP-seq data!
#merge with binding data
lcl_fib_results_binding <- merge(x=lcl_fib_results, y=zfx_binding, by=0, all.x=TRUE)

#Investigate ZFX binding in these results.
fib_results_binding <- lcl_fib_results_binding[! is.na(lcl_fib_results_binding$fib_padj_X),]
fib_results_sig_x_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05,]
fib_results_sig_x_up_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05 & fib_results_binding$fib_log2FC_X > 0,]
fib_results_sig_x_down_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05 & fib_results_binding$fib_log2FC_X < 0,]
fib_results_notsig_x_binding <- fib_results_binding[fib_results_binding$fib_padj_X >= 0.05,]

myList_fib <- list("X_up" = fib_results_sig_x_up_binding$pct_peaks, "notSig_x" = fib_results_notsig_x_binding$pct_peaks, "X_down" = fib_results_sig_x_down_binding$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("X_up","notSig_x","X_down"))

pdf(file="ZFX_binding_fib_Xresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="Fibroblasts",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "X_up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_x","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "X_down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_x","value"])


fib_results_sig_y_binding <- fib_results_binding[fib_results_binding$fib_padj_Y < 0.05,]
fib_results_sig_y_up_binding <- fib_results_binding[fib_results_binding$fib_padj_Y < 0.05 & fib_results_binding$fib_log2FC_Y > 0,]
fib_results_sig_y_down_binding <- fib_results_binding[fib_results_binding$fib_padj_Y < 0.05 & fib_results_binding$fib_log2FC_Y < 0,]
fib_results_notsig_y_binding <- fib_results_binding[fib_results_binding$fib_padj_Y >= 0.05,]

myList_fib <- list("Y_up" = fib_results_sig_y_up_binding$pct_peaks, "notSig_y" = fib_results_notsig_y_binding$pct_peaks, "Y_down" = fib_results_sig_y_down_binding$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("Y_up","notSig_y","Y_down"))

pdf(file="ZFX_binding_fib_Yresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="Fibroblasts",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Y_up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_y","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Y_down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_y","value"])

fib_results_sig_x_y_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05 & fib_results_binding$fib_padj_Y < 0.05,]
fib_results_sig_x_y_up_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05 & fib_results_binding$fib_padj_Y < 0.05 & fib_results_binding$fib_log2FC_X > 0,]
fib_results_sig_x_y_down_binding <- fib_results_binding[fib_results_binding$fib_padj_X < 0.05 & fib_results_binding$fib_padj_Y < 0.05  & fib_results_binding$fib_log2FC_X < 0,]
fib_results_notSig_x_or_y_binding <- fib_results_binding[fib_results_binding$fib_padj_X >= 0.05 & fib_results_binding$fib_padj_Y >= 0.05,]

myList_fib <- list("XY_up" = fib_results_sig_x_y_up_binding$pct_peaks, "notSig_XY" = fib_results_notSig_x_or_y_binding$pct_peaks, "XY_down" = fib_results_sig_x_y_down_binding$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("XY_up","notSig_XY","XY_down"))

pdf(file="ZFX_binding_fib_XYresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="Fibroblasts",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "XY_up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_XY","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "XY_down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "notSig_XY","value"])


#LCLs
lcl_results_binding <- lcl_fib_results_binding[! is.na(lcl_fib_results_binding$lcl_padj_X),]
lcl_results_sig_x_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05,]
lcl_results_sig_x_up_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05 & lcl_results_binding$lcl_log2FC_X > 0,]
lcl_results_sig_x_down_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05 & lcl_results_binding$lcl_log2FC_X < 0,]
lcl_results_notsig_x_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X >= 0.05,]

myList_lcl <- list("X_up" = lcl_results_sig_x_up_binding$pct_peaks, "notSig_x" = lcl_results_notsig_x_binding$pct_peaks, "X_down" = lcl_results_sig_x_down_binding$pct_peaks)

myList_lcl_melt <- melt(myList_lcl)
myList_lcl_melt$L1 <- factor(x=myList_lcl_melt$L1, levels = c("X_up","notSig_x","X_down"))

pdf(file="ZFX_binding_lcl_Xresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_lcl_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "X_up","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_x","value"])
wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "X_down","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_x","value"])

t.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "X_up","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_x","value"])
t.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "X_down","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_x","value"])


lcl_results_sig_y_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_Y < 0.05,]
lcl_results_sig_y_up_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_Y < 0.05 & lcl_results_binding$lcl_log2FC_Y > 0,]
lcl_results_sig_y_down_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_Y < 0.05 & lcl_results_binding$lcl_log2FC_Y < 0,]
lcl_results_notsig_y_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_Y >= 0.05,]

myList_lcl <- list("Y_up" = lcl_results_sig_y_up_binding$pct_peaks, "notSig_y" = lcl_results_notsig_y_binding$pct_peaks, "Y_down" = lcl_results_sig_y_down_binding$pct_peaks)

myList_lcl_melt <- melt(myList_lcl)
myList_lcl_melt$L1 <- factor(x=myList_lcl_melt$L1, levels = c("Y_up","notSig_y","Y_down"))

pdf(file="ZFX_binding_lcl_Yresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_lcl_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "Y_up","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_y","value"])
wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "Y_down","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_y","value"])


lcl_results_sig_x_y_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05 & lcl_results_binding$lcl_padj_Y < 0.05,]
lcl_results_sig_x_y_up_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05 & lcl_results_binding$lcl_padj_Y < 0.05 & lcl_results_binding$lcl_log2FC_X > 0,]
lcl_results_sig_x_y_down_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X < 0.05 & lcl_results_binding$lcl_padj_Y < 0.05  & lcl_results_binding$lcl_log2FC_X < 0,]
lcl_results_notSig_x_or_y_binding <- lcl_results_binding[lcl_results_binding$lcl_padj_X >= 0.05 & lcl_results_binding$lcl_padj_Y >= 0.05,]

myList_lcl <- list("XY_up" = lcl_results_sig_x_y_up_binding$pct_peaks, "notSig_XY" = lcl_results_notSig_x_or_y_binding$pct_peaks, "XY_down" = lcl_results_sig_x_y_down_binding$pct_peaks)

myList_lcl_melt <- melt(myList_lcl)
myList_lcl_melt$L1 <- factor(x=myList_lcl_melt$L1, levels = c("XY_up","notSig_XY","XY_down"))

pdf(file="ZFX_binding_lcl_XYresp.pdf", height=1.75, width=3.3)
ggplot(data=myList_lcl_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "XY_up","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_XY","value"])
wilcox.test(x=myList_lcl_melt[myList_lcl_melt$L1 == "XY_down","value"], y=myList_lcl_melt[myList_lcl_melt$L1 == "notSig_XY","value"])

```

# Processing raw data for CRISPRi experiment
```{r}
# library(tximport)
# annofile <-read.delim(file=paste0(myPath,"Annotations/annotation_with_ERCC.txt"),sep = " ")
# tx2gene <-data.frame("TXNAME" = annofile$transcript_id,"GENEID" = annofile$gene_name)

#Female single knockdowns only
# txi_zfx_zfy_F_files_skd <- as.character(x=metadata_ZFX_Y_F_skd$kallisto_path)
# names(txi_zfx_zfy_F_files_skd) <- metadata_ZFX_Y_F_skd$sample
# txi_zfx_zfy_F_skd <- tximport(txi_zfx_zfy_F_files_skd, type = "kallisto", tx2gene = tx2gene)
# save(txi_zfx_zfy_F_skd,file="txi_zfx_zfy_skd_F.rda")
load(file="txi_zfx_zfy_skd_F.rda")
# write.table(x=txi_zfx_zfy_F_skd$counts, file="txi_zfx_zfy_F_skd_counts.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)
# write.table(x=txi_zfx_zfy_F_skd$abundance, file="txi_zfx_zfy_F_skd_TPM.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)


#Male single knockdowns samples only
# txi_zfx_zfy_M_files_skd <- as.character(x=metadata_ZFX_Y_M_skd$kallisto_path)
# names(txi_zfx_zfy_M_files_skd) <- metadata_ZFX_Y_M_skd$sample
# txi_zfx_zfy_M_skd <- tximport(txi_zfx_zfy_M_files_skd, type = "kallisto", tx2gene = tx2gene)
# save(txi_zfx_zfy_M_skd,file="txi_zfx_zfy_skd_M.rda")
load(file="txi_zfx_zfy_skd_M.rda")
# write.table(x=txi_zfx_zfy_M_skd$counts, file="txi_zfx_zfy_M_skd_counts.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)
# write.table(x=txi_zfx_zfy_M_skd$abundance, file="txi_zfx_zfy_M_skd_TPM.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)


#Double knockdowns
# txi_zfx_zfy_M_files_dkd <- as.character(x=metadata_dkd$kallisto_path)
# names(txi_zfx_zfy_M_files_dkd) <- metadata_dkd$sample
# txi_zfx_zfy_M_dkd <- tximport(txi_zfx_zfy_M_files_dkd, type = "kallisto", tx2gene = tx2gene)
# save(txi_zfx_zfy_M_dkd,file="txi_zfx_zfy_dkd_M.rda")
load(file="txi_zfx_zfy_dkd_M.rda")
# write.table(x=txi_zfx_zfy_M_dkd$counts, file="txi_zfx_zfy_M_dkd_counts.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)
# write.table(x=txi_zfx_zfy_M_dkd$abundance, file="txi_zfx_zfy_M_dkd_TPM.txt", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)
```

## Run DESEq
```{r}
zfx_zfy_F_skd_counts <- read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/txi_zfx_zfy_F_skd_counts.txt"), check.names = FALSE)
zfx_zfy_M_skd_counts <- read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/txi_zfx_zfy_M_skd_counts.txt"), check.names = FALSE)
zfx_zfy_M_dkd_counts <- read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/txi_zfx_zfy_M_dkd_counts.txt"), check.names = FALSE)

suppressPackageStartupMessages(library("DESeq2"))
#If you can generate the tximport files, use DESeqDataSetFromTximport()

#XX skd only, correct for cell line
# dds_zfxCrispri_female_skd <- DESeqDataSetFromMatrix(round(zfx_zfy_F_skd_counts), colData = metadata_ZFX_Y_F_skd, design = ~ genotype + cell_line)
# dds_zfxCrispri_female_skd <- DESeq(dds_zfxCrispri_female_skd)
# save(dds_zfxCrispri_female_skd, file="dds_zfxCrispri_female_skd.rda")
load("dds_zfxCrispri_female_skd.rda")
norm_counts_ZFX_Y_skd_F <- counts(dds_zfxCrispri_female_skd, normalized=TRUE)

#XY skd only, correct for cell line
# dds_zfxCrispri_male_skd <- DESeqDataSetFromMatrix(round(zfx_zfy_M_skd_counts), colData = metadata_ZFX_Y_M_skd, design = ~ genotype + cell_line)
# dds_zfxCrispri_male_skd <- DESeq(dds_zfxCrispri_male_skd)
# save(dds_zfxCrispri_male_skd, file="dds_zfxCrispri_male_skd.rda")
load("dds_zfxCrispri_male_skd.rda")
norm_counts_ZFX_Y_skd_M <- counts(dds_zfxCrispri_male_skd, normalized=TRUE)

#XY dkd only, correct for cell line
# dds_zfxCrispri_male_dkd <- DESeqDataSetFromMatrix(round(zfx_zfy_M_dkd_counts), colData = metadata_dkd, design = ~ genotype + cell_line)
# dds_zfxCrispri_male_dkd <- DESeq(dds_zfxCrispri_male_dkd)
# save(dds_zfxCrispri_male_dkd, file="dds_zfxCrispri_male_dkd.rda")
load("dds_zfxCrispri_male_dkd.rda")
norm_counts_ZFX_Y_dkd_M <- counts(dds_zfxCrispri_male_dkd, normalized = TRUE)
```

#Plotting gene knockdown results, relative expression
```{r}
#First, normalize the counts to remove effects of cell line!
vst_zfxCrispri_skd_xx <- vst(dds_zfxCrispri_female_skd, blind=FALSE)
vst_zfxCrispri_skd_xy <- vst(dds_zfxCrispri_male_skd, blind=FALSE)
vst_zfxCrispri_dkd <- vst(dds_zfxCrispri_male_dkd, blind=FALSE)

vst_normCounts_skd_XX <- 2^(assay(vst_zfxCrispri_skd_xx))
vst_normCounts_skd_XY <- 2^(assay(vst_zfxCrispri_skd_xy))
vst_normCounts_dkd <- 2^(assay(vst_zfxCrispri_dkd))

myOrange <- "#d95f0295"
myOrange_light <- "#d95f0250"

myPurple <- "#7570B395"

pdf(file="XX_ZFX_skd.pdf", width=1, height=1.5)
#First ZFX knockdown in XX cells:
XX_zfx_kd <- data.frame("ZFX" = vst_normCounts_skd_XX["ZFX",], row.names = colnames(vst_normCounts_skd_XX))
data_pull <- metadata_ZFX_Y_F_skd[,c("sample","karyotype","genotype", "batch_libprep")]
rownames(data_pull) <- data_pull$sample
plot_data_XX <- cbind(data_pull, XX_zfx_kd)

#normalize based on control median
control_median <- median(plot_data_XX[plot_data_XX$genotype == "Control", "ZFX"])
plot_data_XX$ZFX <- plot_data_XX$ZFX / control_median

#Make plot
ggplot(plot_data_XX, aes(x=genotype, y=ZFX, col=genotype)) + 
    geom_jitter(position = position_jitter(0.1), stroke=0) +
    scale_color_manual(values=c("#00000040",myOrange)) +
      geom_violin(color = "black", fill="#00000000", size=0.25) +
  stat_summary(fun.data=data_summary, size=0.25, color="black") +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank(),
       legend.position="none"
      ) + labs(y="Relative expression") + ylim(0,1.2)

dev.off()


pdf(file="XY_ZFX_skd.pdf", width=1.2, height=1.5)
XY_zfx_kd <- data.frame("ZFX" = vst_normCounts_skd_XY["ZFX",], row.names = colnames(vst_normCounts_skd_XY))
data_pull <- metadata_ZFX_Y_M_skd[,c("sample","karyotype","genotype", "batch_libprep")]
rownames(data_pull) <- data_pull$sample
plot_data_XY <- cbind(data_pull, XY_zfx_kd)

#normalize based on control median
control_median <- median(plot_data_XY[plot_data_XY$genotype == "Control", "ZFX"])
plot_data_XY$ZFX <- plot_data_XY$ZFX / control_median

plot_data_XY$genotype <- factor(x=plot_data_XY$genotype, levels = c("Control","ZFX_kd","ZFY_kd"))

#Make plot
ggplot(plot_data_XY, aes(x=genotype, y=ZFX, col=genotype)) + 
  geom_jitter(position = position_jitter(0.1), stroke=0) +
  scale_color_manual(values=c("#00000040",myOrange, myPurple)) +  
  geom_violin(color = "black", fill="#00000000", size=0.25) +
    stat_summary(fun.data=data_summary, size=0.25, color="black") +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank(),
       legend.position="none"
      ) + labs(y="Relative expression") + ylim(0,1.2)

dev.off()


pdf(file="XY_ZFY_skd.pdf", width=1.2, height=1.5)
XY_zfy_kd <- data.frame("ZFY" = vst_normCounts_skd_XY["ZFY",], row.names = colnames(vst_normCounts_skd_XY))
data_pull <- metadata_ZFX_Y_M_skd[,c("sample","karyotype","genotype", "batch_libprep")]
rownames(data_pull) <- data_pull$sample
plot_data_XY <- cbind(data_pull, XY_zfy_kd)

#normalize based on control median
control_median <- median(plot_data_XY[plot_data_XY$genotype == "Control", "ZFY"])
plot_data_XY$ZFY <- plot_data_XY$ZFY / control_median

plot_data_XY$genotype <- factor(x=plot_data_XY$genotype, levels = c("Control","ZFX_kd","ZFY_kd"))

#Make plot
ggplot(plot_data_XY, aes(x=genotype, y=ZFY, col=genotype)) + 
    geom_jitter(position = position_jitter(0.1), stroke=0) +
    scale_color_manual(values=c("#00000040",myOrange, myPurple)) +
  geom_violin(color = "black", fill="#00000000", size=0.25) +
    stat_summary(fun.data=data_summary, size=0.25, color="black") +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank(),
       legend.position="none"
      ) + labs(y="Relative expression") + ylim(0,1.2)

dev.off()

#DOUBLE KD
pdf(file="XY_ZFX_dkd.pdf",width=1, height=1.5)
XY_zfy_kd <- data.frame("ZFX" = vst_normCounts_dkd["ZFX",], row.names = colnames(vst_normCounts_dkd))
data_pull <- metadata_dkd[,c("sample","karyotype","genotype", "batch_libprep")]
rownames(data_pull) <- data_pull$sample
plot_data_XY <- cbind(data_pull, XY_zfy_kd)

#normalize based on control median
control_median <- median(plot_data_XY[plot_data_XY$genotype == "Control", "ZFX"])
plot_data_XY$ZFX <- plot_data_XY$ZFX / control_median

plot_data_XY$genotype <- factor(x=plot_data_XY$genotype, levels = c("Control","ZFX_ZFY_kd"))

#Make plot
ggplot(plot_data_XY, aes(x=genotype, y=ZFX, col=genotype)) + 
    geom_jitter(position = position_jitter(0.1), stroke=0) +
    scale_color_manual(values=c("#00000040","black")) +
  geom_violin(color = "black", fill="#00000000", size=0.25) +
    stat_summary(fun.data=data_summary, size=0.25, color="black") +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank(),
       legend.position="none"
      ) + labs(y="Relative expression") + ylim(0,1.2)

dev.off()


pdf(file="XY_ZFY_dkd.pdf",width=1, height=1.5)
XY_zfy_kd <- data.frame("ZFY" = vst_normCounts_dkd["ZFY",], row.names = colnames(vst_normCounts_dkd))
data_pull <- metadata_dkd[,c("sample","karyotype","genotype", "batch_libprep")]
rownames(data_pull) <- data_pull$sample
plot_data_XY <- cbind(data_pull, XY_zfy_kd)

#normalize based on control median
control_median <- median(plot_data_XY[plot_data_XY$genotype == "Control", "ZFY"])
plot_data_XY$ZFY <- plot_data_XY$ZFY / control_median

plot_data_XY$genotype <- factor(x=plot_data_XY$genotype, levels = c("Control","ZFX_ZFY_kd"))

#Make plot
ggplot(plot_data_XY, aes(x=genotype, y=ZFY, col=genotype)) + 
    geom_jitter(position = position_jitter(0.1), stroke=0) +
    scale_color_manual(values=c("#00000040","black")) +
  geom_violin(color = "black", fill="#00000000", size=0.25) +
    stat_summary(fun.data=data_summary, size=0.25, color="black") +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.title.x = element_blank(),
       legend.position="none"
      ) + labs(y="Relative expression") + ylim(0,1.2)

dev.off()
```


# Analyze differential expn
```{r}
#bring in fib exp genes
load(file=paste0(myPath,"Linear_regressions/Fibroblasts/fib_expressed.rda"))

#female samp analysis control vs ZFX
myList <- matrix(c("ZFX_kd","Control"),ncol=2, byrow = TRUE)
deseqFactorAnalysis(deseqObject = dds_zfxCrispri_female_skd, myVariable = "genotype", comparisonList = myList, expressedGeneList = fib_expressed$expressedGenes, p_value=0.05, myTitle = "XX_skd")

#male samp analysis control vs ZFX; control vs ZFY
myList <- matrix(c("ZFX_kd","Control","ZFY_kd","Control"),ncol=2, byrow = TRUE)
deseqFactorAnalysis(deseqObject = dds_zfxCrispri_male_skd, myVariable = "genotype", comparisonList = myList, expressedGeneList = fib_expressed$expressedGenes, p_value=0.05, myTitle = "XY_skd") 

normCounts_XY_singleKDs <- counts(dds_zfxCrispri_male_skd, normalized=TRUE)
write.table(x=normCounts_XY_singleKDs, file="normCounts_XY_singleKDs.txt", quote=FALSE, sep="\t", col.names = TRUE, row.names = TRUE)

myList <- matrix(c("ZFX_ZFY_kd","Control"),ncol=2, byrow = TRUE)
deseqFactorAnalysis(deseqObject = dds_zfxCrispri_male_dkd, myVariable = "genotype", comparisonList = myList, expressedGeneList = fib_expressed$expressedGenes, p_value=0.05, myTitle = "XY_dkd") 

normCounts_XY_doubleKD <- counts(dds_zfxCrispri_male_dkd, normalized=TRUE)
write.table(x=normCounts_XY_doubleKD, file="normCounts_XY_doubleKD.txt", quote=FALSE, sep="\t", col.names = TRUE, row.names = TRUE)

```

##XX ZFX kd vs Control
```{r}
#XX Control vs ZFX kd
female_res_zfx_cntrl_exp <- na.omit(read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/XX_skd_ZFX_kdvsControl_DESeq_expressed.txt"), stringsAsFactors = FALSE))
#merge with binding data
female_res_zfx_cntrl_exp_binding <- merge(x=female_res_zfx_cntrl_exp, y=zfx_binding, by.x=0, by.y=0, all.x=TRUE)

female_res_zfx_cntrl_sig <- female_res_zfx_cntrl_exp_binding[female_res_zfx_cntrl_exp_binding$padj < 0.05,] 
dim(female_res_zfx_cntrl_sig) #4730 significant genes
female_res_zfx_cntrl_notsig <- female_res_zfx_cntrl_exp_binding[female_res_zfx_cntrl_exp_binding$padj >= 0.05,] 
dim(female_res_zfx_cntrl_notsig) #7263 not significant genes
female_res_zfx_cntrl_sig_up <- female_res_zfx_cntrl_sig[female_res_zfx_cntrl_sig$log2FoldChange > 0,] 
dim(female_res_zfx_cntrl_sig_up) #2414 significant up genes
female_res_zfx_cntrl_sig_down <- female_res_zfx_cntrl_sig[female_res_zfx_cntrl_sig$log2FoldChange < 0,] 
dim(female_res_zfx_cntrl_sig_down) #2316 significant down genes

female_res_zfx_cntrl_exp_auto <- female_res_zfx_cntrl_exp_binding[female_res_zfx_cntrl_exp_binding$Gene %in% autosome_genes,]
female_res_zfx_cntrl_sig_auto <- female_res_zfx_cntrl_sig[female_res_zfx_cntrl_sig$Row.names %in% autosome_genes,]
dim(female_res_zfx_cntrl_sig_auto) #4572 significant autosomal genes
female_res_zfx_cntrl_sig_sexChrom <- female_res_zfx_cntrl_sig[! rownames(female_res_zfx_cntrl_sig) %in% autosome_genes,]
dim(female_res_zfx_cntrl_sig_sexChrom) #158 significant sex chrom genes

#break up into up and down
female_res_zfx_cntrl_auto_sigup <- female_res_zfx_cntrl_sig_auto[female_res_zfx_cntrl_sig_auto$log2FoldChange > 0,]
dim(female_res_zfx_cntrl_auto_sigup)
write.table(x=female_res_zfx_cntrl_auto_sigup$Gene, file = "XX_zfx_v_cntrl_up_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)
female_res_zfx_cntrl_auto_sigdown <- female_res_zfx_cntrl_sig_auto[female_res_zfx_cntrl_sig_auto$log2FoldChange < 0,]
dim(female_res_zfx_cntrl_auto_sigdown)
write.table(x=female_res_zfx_cntrl_auto_sigdown$Gene, file = "XX_zfx_v_cntrl_down_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

female_res_zfx_cntrl_sexChrom_sigup <- female_res_zfx_cntrl_sig_sexChrom[female_res_zfx_cntrl_sig_sexChrom$log2FoldChange > 0,]
dim(female_res_zfx_cntrl_sexChrom_sigup)
female_res_zfx_cntrl_sexChrom_sigdown <- female_res_zfx_cntrl_sig_sexChrom[female_res_zfx_cntrl_sig_sexChrom$log2FoldChange < 0,]
dim(female_res_zfx_cntrl_sexChrom_sigdown)


pdf(file="XX_Cntrl_v_ZFX_volcano.pdf", width=1.75, height=1.75)
#Make volcano plot
ggplot() +
  geom_point(data = female_res_zfx_cntrl_exp_binding[female_res_zfx_cntrl_exp_binding$padj >= 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color="#00000040", stroke=0) +
  geom_point(data = female_res_zfx_cntrl_exp_binding[female_res_zfx_cntrl_exp_binding$padj < 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color=myOrange_light, stroke=0) +
  geom_vline(xintercept = 0, lty=2) +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
       legend.position="none"
      ) + labs(y="-log10(FDR)", x="ZFXkd vs Control\n log2 fold-change") + ylim(0,175) + scale_x_continuous(limits = c(-4, 1.5), breaks = seq(-4, 1.5, by = 1))
dev.off()


#Investigate ZFX binding in these results.
female_res_zfx_cntrl_sig_bound <- female_res_zfx_cntrl_sig[is.na(female_res_zfx_cntrl_sig$sum_peaks),] 
dim(female_res_zfx_cntrl_sig_bound) #3559 of 4730 significant genes are bound by ZFX in at least one cell type; 3055 in at least two cell types; 2571 in three; 1991 in four
female_res_zfx_cntrl_sig_up_bound <- female_res_zfx_cntrl_sig_up[! is.na(female_res_zfx_cntrl_sig_up$sum_peaks) & (female_res_zfx_cntrl_sig_up$sum_peaks > 3),] 
dim(female_res_zfx_cntrl_sig_up_bound) #2151 of 2414 significant up genes are bound by ZFX at TSS in at least one cell type; 1989 in at least two cell types; 1804 in three; 1506 in four; 192 NA...
female_res_zfx_cntrl_sig_down_bound <- female_res_zfx_cntrl_sig_down[is.na(female_res_zfx_cntrl_sig_down$sum_peaks),] 
dim(female_res_zfx_cntrl_sig_down_bound) #1408 of 2316 significant down genes are bound by ZFX at TSS in at least one cell type; 1066 in at least two cell types; 767 in three; 485 in four; 113 na

female_res_zfx_cntrl_notsig_bound <- female_res_zfx_cntrl_notsig[is.na(female_res_zfx_cntrl_notsig$sum_peaks) ,] 
dim(female_res_zfx_cntrl_notsig_bound) #1284 of 7263 not significant genes have no ZFX binding; 731 with one binding site; 845 with 2; 1103 with 3; 2976 with 4; 318 NA

#Binding analysis!
myList_fib <- list("Up" = female_res_zfx_cntrl_sig_up$pct_peaks, "NotSig" = female_res_zfx_cntrl_notsig$pct_peaks, "Down" = female_res_zfx_cntrl_sig_down$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("Up","NotSig","Down"))

pdf(file="ZFX_binding_fib_ZFXvControlCRISPRi_XX.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
 # geom_hline(yintercept = 0, size=0.25, lty=2, col="#00000040") + 
   # geom_hline(yintercept = 1, size=0.25, lty=2, col="#00000040") + 
  # geom_jitter(width=0.2,stroke=0, color="#00000080") +
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="ZFX vs Control, XX",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
```

##XY ZFX kd vs Control
```{r}
male_skd_res_zfx_cntrl_exp <- na.omit(read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/XY_skd_ZFX_kdvsControl_DESeq_expressed.txt"), stringsAsFactors = FALSE))
#merge with binding data
male_skd_res_zfx_cntrl_exp_binding <- merge(x=male_skd_res_zfx_cntrl_exp, y=zfx_binding, by.x=0, by.y=0, all.x=TRUE)
male_skd_res_zfx_cntrl_sig <- male_skd_res_zfx_cntrl_exp_binding[male_skd_res_zfx_cntrl_exp_binding$padj < 0.05,]
dim(male_skd_res_zfx_cntrl_sig)
male_skd_res_zfx_cntrl_notsig <- male_skd_res_zfx_cntrl_exp_binding[male_skd_res_zfx_cntrl_exp_binding$padj >= 0.05,]
dim(male_skd_res_zfx_cntrl_notsig)
male_skd_res_zfx_cntrl_sig_up <- male_skd_res_zfx_cntrl_sig[male_skd_res_zfx_cntrl_sig$log2FoldChange > 0,]
dim(male_skd_res_zfx_cntrl_sig_up)
male_skd_res_zfx_cntrl_sig_down <- male_skd_res_zfx_cntrl_sig[male_skd_res_zfx_cntrl_sig$log2FoldChange < 0,]
dim(male_skd_res_zfx_cntrl_sig_down)

male_skd_res_zfx_cntrl_exp_auto <- male_skd_res_zfx_cntrl_exp_binding[male_skd_res_zfx_cntrl_exp_binding$Row.names %in% autosome_genes,]
male_skd_res_zfx_cntrl_sig_auto <- male_skd_res_zfx_cntrl_exp_auto[male_skd_res_zfx_cntrl_exp_auto$padj < 0.05,]
dim(male_skd_res_zfx_cntrl_sig_auto)

#break up into up and down
male_skd_res_zfx_cntrl_sig_auto_up <- male_skd_res_zfx_cntrl_sig_auto[male_skd_res_zfx_cntrl_sig_auto$log2FoldChange > 0,]
dim(male_skd_res_zfx_cntrl_sig_auto_up)
write.table(x=male_skd_res_zfx_cntrl_sig_auto_up$Gene, file = "XY_zfx_v_cntrl_up_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

male_skd_res_zfx_cntrl_sig_auto_down <- male_skd_res_zfx_cntrl_sig_auto[male_skd_res_zfx_cntrl_sig_auto$log2FoldChange < 0,]
dim(male_skd_res_zfx_cntrl_sig_auto_down)
write.table(x=male_skd_res_zfx_cntrl_sig_auto_down$Gene, file = "XY_zfx_v_cntrl_down_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

male_skd_res_zfx_cntrl_exp_sexChrom <- male_skd_res_zfx_cntrl_exp[! rownames(male_skd_res_zfx_cntrl_exp) %in% autosome_genes,]
male_skd_res_zfx_cntrl_sig_sexChrom <- male_skd_res_zfx_cntrl_exp_sexChrom[male_skd_res_zfx_cntrl_exp_sexChrom$padj < 0.05,]
dim(male_skd_res_zfx_cntrl_sig_sexChrom)

#break up into up and down
male_skd_res_zfx_cntrl_sig_sexChrom_up <- male_skd_res_zfx_cntrl_sig_sexChrom[male_skd_res_zfx_cntrl_sig_sexChrom$log2FoldChange > 0,]
dim(male_skd_res_zfx_cntrl_sig_sexChrom_up)
male_skd_res_zfx_cntrl_sig_sexChrom_down <- male_skd_res_zfx_cntrl_sig_sexChrom[male_skd_res_zfx_cntrl_sig_sexChrom$log2FoldChange < 0,]
dim(male_skd_res_zfx_cntrl_sig_sexChrom_down)

pdf(file="XY_Cntrl_v_ZFX_volcano.pdf", width=1.75, height=1.75)
#Make volcano plot
ggplot() +
  geom_point(data = male_skd_res_zfx_cntrl_exp[male_skd_res_zfx_cntrl_exp$padj >= 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color="#00000040", stroke=0) +
  geom_point(data = male_skd_res_zfx_cntrl_exp[male_skd_res_zfx_cntrl_exp$padj < 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color=myOrange_light, stroke=0) +
  geom_vline(xintercept = 0, lty=2) +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
       legend.position="none"
      ) + 
  labs(y="-log10(FDR)", x="Control vs ZFXkd\n log2 fold-change") +
 ylim(0,165) + scale_x_continuous(limits = c(-4, 1.5), breaks = seq(-4, 1.5, by = 1))
dev.off()


#Binding analysis!
myList_fib <- list("Up" = male_skd_res_zfx_cntrl_sig_up$pct_peaks, "NotSig" = male_skd_res_zfx_cntrl_notsig$pct_peaks, "Down" = male_skd_res_zfx_cntrl_sig_down$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("Up","NotSig","Down"))

pdf(file="ZFX_binding_fib_ZFXvControlCRISPRi_XY.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
 # geom_hline(yintercept = 0, size=0.25, lty=2, col="#00000040") + 
   # geom_hline(yintercept = 1, size=0.25, lty=2, col="#00000040") + 
  # geom_jitter(width=0.2,stroke=0, color="#00000080") +
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="ZFX vs Control, XY",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
```

##XY ZFY kd vs control
```{r}
#XY Control vs ZFY kd
male_skd_res_zfy_cntrl_exp <- na.omit(read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/XY_skd_ZFY_kdvsControl_DESeq_expressed.txt"), stringsAsFactors = FALSE))
#merge with binding data
male_skd_res_zfy_cntrl_exp_binding <- merge(x=male_skd_res_zfy_cntrl_exp, y=zfx_binding, by.x=0, by.y=0, all.x=TRUE)

male_skd_res_zfy_cntrl_sig <- male_skd_res_zfy_cntrl_exp_binding[male_skd_res_zfy_cntrl_exp_binding$padj < 0.05,]
dim(male_skd_res_zfy_cntrl_sig)
male_skd_res_zfy_cntrl_notsig <- male_skd_res_zfy_cntrl_exp_binding[male_skd_res_zfy_cntrl_exp_binding$padj >= 0.05,]
dim(male_skd_res_zfy_cntrl_notsig)
male_skd_res_zfy_cntrl_sig_up <- male_skd_res_zfy_cntrl_sig[male_skd_res_zfy_cntrl_sig$log2FoldChange > 0,]
dim(male_skd_res_zfy_cntrl_sig_up)
male_skd_res_zfy_cntrl_sig_down <- male_skd_res_zfy_cntrl_sig[male_skd_res_zfy_cntrl_sig$log2FoldChange < 0,]
dim(male_skd_res_zfy_cntrl_sig_down)

male_skd_res_zfy_cntrl_exp_auto <- male_skd_res_zfy_cntrl_exp_binding[male_skd_res_zfy_cntrl_exp_binding$Row.names %in% autosome_genes,]
male_skd_res_zfy_cntrl_exp_auto_sig <- male_skd_res_zfy_cntrl_exp_auto[male_skd_res_zfy_cntrl_exp_auto$padj < 0.05,]
dim(male_skd_res_zfy_cntrl_exp_auto_sig)

male_skd_res_zfy_cntrl_exp_auto_sig_up <- male_skd_res_zfy_cntrl_exp_auto_sig[male_skd_res_zfy_cntrl_exp_auto_sig$log2FoldChange > 0,]
dim(male_skd_res_zfy_cntrl_exp_auto_sig_up)
write.table(x=male_skd_res_zfy_cntrl_exp_auto_sig_up$Gene, file = "XY_zfy_v_cntrl_up_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

male_skd_res_zfy_cntrl_exp_auto_sig_down <- male_skd_res_zfy_cntrl_exp_auto_sig[male_skd_res_zfy_cntrl_exp_auto_sig$log2FoldChange < 0,]
dim(male_skd_res_zfy_cntrl_exp_auto_sig_down)
write.table(x=male_skd_res_zfy_cntrl_exp_auto_sig_down$Gene, file = "XY_zfy_v_cntrl_down_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

male_skd_res_zfy_cntrl_exp_sexChr <- male_skd_res_zfy_cntrl_exp[!rownames(male_skd_res_zfy_cntrl_exp) %in% autosome_genes,]
male_skd_res_zfy_cntrl_exp_sexChr_sig <- male_skd_res_zfy_cntrl_exp_sexChr[male_skd_res_zfy_cntrl_exp_sexChr$padj < 0.05,]
dim(male_skd_res_zfy_cntrl_exp_sexChr_sig)

male_skd_res_zfy_cntrl_exp_sexChr_sig_up <- male_skd_res_zfy_cntrl_exp_sexChr_sig[male_skd_res_zfy_cntrl_exp_sexChr_sig$log2FoldChange > 0,]
dim(male_skd_res_zfy_cntrl_exp_sexChr_sig_up)
male_skd_res_zfy_cntrl_exp_sexChr_sig_down <- male_skd_res_zfy_cntrl_exp_sexChr_sig[male_skd_res_zfy_cntrl_exp_sexChr_sig$log2FoldChange < 0,]
dim(male_skd_res_zfy_cntrl_exp_sexChr_sig_down)

pdf(file="XY_Cntrl_v_ZFY_volcano_lims.pdf", width=1.75, height=1.75)
#Make volcano plot
ggplot() +
  geom_point(data = male_skd_res_zfy_cntrl_exp[male_skd_res_zfy_cntrl_exp$padj >= 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color="#00000040", stroke=0) +
  geom_point(data = male_skd_res_zfy_cntrl_exp[male_skd_res_zfy_cntrl_exp$padj < 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color=myPurple, stroke=0) +
  geom_vline(xintercept = 0, lty=2) +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
       legend.position="none"
      )  + labs(y="-log10(FDR)", x="Control vs ZFYkd\n log2 fold-change") + ylim(0,10) + scale_x_continuous(limits = c(-4, 1.5), breaks = seq(-4, 1.5, by = 1))
dev.off()

#Binding analysis!
myList_fib <- list("Up" = male_skd_res_zfy_cntrl_sig_up$pct_peaks, "NotSig" = male_skd_res_zfy_cntrl_notsig$pct_peaks, "Down" = male_skd_res_zfy_cntrl_sig_down$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("Up","NotSig","Down"))

pdf(file="ZFX_binding_fib_ZFYvControlCRISPRi_XY.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
 # geom_hline(yintercept = 0, size=0.25, lty=2, col="#00000040") + 
   # geom_hline(yintercept = 1, size=0.25, lty=2, col="#00000040") + 
  # geom_jitter(width=0.2,stroke=0, color="#00000080") +
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="ZFYkd vs Control, XY",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
```

##XY double kd vs control
```{r}
#XY Control vs DKd
male_dkd_res_dko_cntrl_exp <- na.omit(read.delim(file=paste0(myPath,"ZFX_ZFY_CRISPRi/XY_dkd_ZFX_ZFY_kdvsControl_DESeq_expressed.txt"), stringsAsFactors = FALSE))
#merge with binding data
male_dkd_res_dko_cntrl_exp_binding <- merge(x=male_dkd_res_dko_cntrl_exp, y=zfx_binding, by.x=0, by.y=0, all.x=TRUE)

male_dkd_res_dko_cntrl_sig <- male_dkd_res_dko_cntrl_exp_binding[male_dkd_res_dko_cntrl_exp_binding$padj < 0.05,]
dim(male_dkd_res_dko_cntrl_sig)
male_dkd_res_dko_cntrl_notsig <- male_dkd_res_dko_cntrl_exp_binding[male_dkd_res_dko_cntrl_exp_binding$padj >= 0.05,]
dim(male_dkd_res_dko_cntrl_notsig)
male_dkd_res_dko_cntrl_sig_up <- male_dkd_res_dko_cntrl_sig[male_dkd_res_dko_cntrl_sig$log2FoldChange > 0,]
dim(male_dkd_res_dko_cntrl_sig_up)
male_dkd_res_dko_cntrl_sig_down <- male_dkd_res_dko_cntrl_sig[male_dkd_res_dko_cntrl_sig$log2FoldChange < 0,]
dim(male_dkd_res_dko_cntrl_sig_down)

male_dkd_res_dko_cntrl_auto <- male_dkd_res_dko_cntrl_exp_binding[male_dkd_res_dko_cntrl_exp_binding$Row.names %in% autosome_genes,]
male_dkd_res_dko_cntrl_auto_sig <- male_dkd_res_dko_cntrl_auto[male_dkd_res_dko_cntrl_auto$padj < 0.05,]
dim(male_dkd_res_dko_cntrl_auto_sig)

male_dkd_res_dko_cntrl_auto_sig_up <- male_dkd_res_dko_cntrl_auto_sig[male_dkd_res_dko_cntrl_auto_sig$log2FoldChange > 0,]
dim(male_dkd_res_dko_cntrl_auto_sig_up)
write.table(x=male_dkd_res_dko_cntrl_auto_sig_up$Gene, file = "XY_zfxzfy_v_cntrl_up_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)

male_dkd_res_dko_cntrl_auto_sig_down <- male_dkd_res_dko_cntrl_auto_sig[male_dkd_res_dko_cntrl_auto_sig$log2FoldChange <0,]
dim(male_dkd_res_dko_cntrl_auto_sig_down)
write.table(x=male_dkd_res_dko_cntrl_auto_sig_down$Gene, file = "XY_zfxzfy_v_cntrl_down_auto_genes.txt", sep="\t", col.names = FALSE, row.names = FALSE, quote=FALSE)


male_dkd_res_dko_cntrl_sexChr <- male_dkd_res_dko_cntrl_exp_binding[! rownames(male_dkd_res_dko_cntrl_exp_binding) %in% autosome_genes,]
male_dkd_res_dko_cntrl_sexChr_sig <- male_dkd_res_dko_cntrl_sexChr[male_dkd_res_dko_cntrl_sexChr$padj < 0.05,]
dim(male_dkd_res_dko_cntrl_sexChr_sig)

male_dkd_res_dko_cntrl_sexChr_sig_up <- male_dkd_res_dko_cntrl_sexChr_sig[male_dkd_res_dko_cntrl_sexChr_sig$log2FoldChange > 0,]
dim(male_dkd_res_dko_cntrl_sexChr_sig_up)
male_dkd_res_dko_cntrl_sexChr_sig_down <- male_dkd_res_dko_cntrl_sexChr_sig[male_dkd_res_dko_cntrl_sexChr_sig$log2FoldChange <0,]
dim(male_dkd_res_dko_cntrl_sexChr_sig_down)

pdf(file="XY_cntrl_v_ZFXZFY_volcano.pdf", width=1.75, height=1.75)
#Make volcano plot
ggplot() +
  geom_point(data = male_dkd_res_dko_cntrl_exp[male_dkd_res_dko_cntrl_exp$padj >= 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color="#00000040", stroke=0) +
  geom_point(data = male_dkd_res_dko_cntrl_exp[male_dkd_res_dko_cntrl_exp$padj < 0.05,], mapping=aes(x=log2FoldChange, y=-log10(padj)), color="#32a85280", stroke=0) +
    geom_text_repel(data = male_dkd_res_dko_cntrl_exp[male_dkd_res_dko_cntrl_exp$Gene %in% c("ZFX","ZFY"),], mapping=aes(x=log2FoldChange, y=-log10(padj), label=Gene), size=2, min.segment.length = 0) +
  geom_vline(xintercept = 0, lty=2) +
  theme_classic(base_size = 8) +
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
       legend.position="none"
      )  + labs(y="-log10(FDR)", x="Control vs ZFX & ZFYkd\n log2 fold-change")  + scale_x_continuous(limits = c(-4, 1.5), breaks = seq(-4, 1.5, by = 1))
dev.off()

#Binding analysis!
myList_fib <- list("Up" = male_dkd_res_dko_cntrl_sig_up$pct_peaks, "NotSig" = male_dkd_res_dko_cntrl_notsig$pct_peaks, "Down" = male_dkd_res_dko_cntrl_sig_down$pct_peaks)

myList_fib_melt <- melt(myList_fib)
myList_fib_melt$L1 <- factor(x=myList_fib_melt$L1, levels = c("Up","NotSig","Down"))

pdf(file="ZFX_binding_fib_dkdvControlCRISPRi_XY.pdf", height=1.75, width=3.3)
ggplot(data=myList_fib_melt, mapping = aes(x=L1, y=value, fill=L1)) + 
 # geom_hline(yintercept = 0, size=0.25, lty=2, col="#00000040") + 
   # geom_hline(yintercept = 1, size=0.25, lty=2, col="#00000040") + 
  # geom_jitter(width=0.2,stroke=0, color="#00000080") +
   geom_violin(col="#00000000",scale = "width", fill="#00000040") +
   stat_summary(fun.data=data_summary, size=0.25) +
ylim(-0.01,1.01) +
theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="ZFX & ZFYkd vs Control, XY",
    y="Pct ZFX binding in 4 cell types", x=""
  ) 
dev.off()

wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Up","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
wilcox.test(x=myList_fib_melt[myList_fib_melt$L1 == "Down","value"], y=myList_fib_melt[myList_fib_melt$L1 == "NotSig","value"])
```

## Make venn diagrams to compare the results --> they are very similar.
```{r}
#Compare ZFX-responsive, ZFY-responsive in XX and XY
#Autosome genes
threeGroupVenn_geneNames(setA = female_res_zfx_cntrl_sig_auto$Gene, setAname = "XX_ZFXkd", setB = male_skd_res_zfx_cntrl_sig_auto$Gene, setBname = "XY_ZFXkd", setC = male_skd_res_zfy_cntrl_exp_auto_sig$Gene, setCname = "XY_ZFYkd", expGenes = fib_expressed$expAutoGenes, filename = "Autosomes_XX_XY_skd.pdf")

threeGroupVenn_geneNames(setA = male_dkd_res_dko_cntrl_auto_sig$Gene, setAname = "XY_ZFX_ZFYkd", setB = male_skd_res_zfx_cntrl_sig_auto$Gene, setBname = "XY_ZFXkd", setC = male_skd_res_zfy_cntrl_exp_auto_sig$Gene, setCname = "XY_ZFYkd", expGenes = fib_expressed$expAutoGenes, filename = "Autosomes_XX_XY_dkd.pdf")


#Correlation plots for ZFX and ZFY-responses in 46,XY cells.
zfx_and_zfy_resp_genes_XY <- intersect(male_skd_res_zfx_cntrl_sig_auto$Gene, male_skd_res_zfy_cntrl_exp_auto_sig$Gene)
myData <- data.frame("x" = male_skd_res_zfx_cntrl_exp[zfx_and_zfy_resp_genes_XY,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[zfx_and_zfy_resp_genes_XY,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[zfx_and_zfy_resp_genes_XY,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[zfx_and_zfy_resp_genes_XY,"lfcSE"])

t.test(x = myData$x, y=myData$y, paired = TRUE)

paper2_corrPlot_deming_err(myData = myData, myTitle = "CorrPlot_ZFXkd_v_ZFYkd_demingErr.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFX kd vs control 46,XY\nlog2 fold change", myYlab = "ZFY kd vs control 46,XY\nlog2 fold change", pointsCol = default_pointsCol, demingBehind = TRUE)
```

#All 3 plots
```{r}
zfx_resp_genes_XY <- setdiff(male_skd_res_zfx_cntrl_sig_auto$Gene, male_skd_res_zfy_cntrl_exp_auto_sig$Gene)
zfy_resp_genes_XY <- setdiff(male_skd_res_zfy_cntrl_exp_auto_sig$Gene, male_skd_res_zfx_cntrl_sig_auto$Gene)

myData_XY <- data.frame("x" = male_skd_res_zfx_cntrl_exp[zfx_and_zfy_resp_genes_XY,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[zfx_and_zfy_resp_genes_XY,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[zfx_and_zfy_resp_genes_XY,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[zfx_and_zfy_resp_genes_XY,"lfcSE"])

myData_Xonly <- data.frame("x" = male_skd_res_zfx_cntrl_exp[zfx_resp_genes_XY,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[zfx_resp_genes_XY,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[zfx_resp_genes_XY,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[zfx_resp_genes_XY,"lfcSE"])

myData_Yonly <- data.frame("x" = male_skd_res_zfx_cntrl_exp[zfy_resp_genes_XY,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[zfy_resp_genes_XY,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[zfy_resp_genes_XY,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[zfy_resp_genes_XY,"lfcSE"])


myLim <- makeSymmetric_X_Y(myData_XY$x, myData_XY$y)
# myLim_Xonly <- makeSymmetric_X_Y(myData_Xonly$x, myData_Xonly$y)
# myLim_Yonly <- makeSymmetric_X_Y(myData_Yonly$x, myData_Yonly$y)

myData_min <- myLim[1] - 0.1
myData_max <- myLim[2] + 0.1
x_range <- seq(myData_min - 1, myData_max + 1, 0.01)

#X and Y calculations
my.deming_err_XY <- deming(formula = myData_XY$y ~ myData_XY$x, xstd = myData_XY$x_err, ystd = myData_XY$y_err) # This works...it just takes FOREVER!
deming.slope_err_XY <- my.deming_err_XY$coefficients[2]
deming.slope_seq_XY <- seq(my.deming_err_XY$ci[2,1], my.deming_err_XY$ci[2,2], 0.001)
deming.int_err_XY <- my.deming_err_XY$coefficients[1]
deming.int_seq_XY <- seq(my.deming_err_XY$ci[1,1], my.deming_err_XY$ci[1,2], 0.0001)
deming.ribbon_table_XY <- data.frame("x" = x_range, "fit" = ((deming.slope_err_XY * x_range) + deming.int_err_XY), "y.min" = numeric(length = length(x_range)), "y.max" = numeric(length=length(x_range)))
deming.95ci.lines_XY <- data.frame("intercept" = rep(deming.int_seq_XY, each = length(deming.slope_seq_XY)), "slope" = deming.slope_seq_XY)
deming.ribbon_table_XY[,c("y.min","y.max")] <- t(sapply(x_range, getMaxMin, mydeming_lines = deming.95ci.lines_XY))
myCor_pear_XY <- cor.test(myData_XY$x, myData_XY$y, method=c("pearson"))


#X only calculations
my.deming_err_Xonly <- deming(formula = myData_Xonly$y ~ myData_Xonly$x, xstd = myData_Xonly$x_err, ystd = myData_Xonly$y_err) # This works...it just takes FOREVER!
deming.slope_err_Xonly <- my.deming_err_Xonly$coefficients[2]
deming.slope_seq_Xonly <- seq(my.deming_err_Xonly$ci[2,1], my.deming_err_Xonly$ci[2,2], 0.001)
deming.int_err_Xonly <- my.deming_err_Xonly$coefficients[1]
deming.int_seq_Xonly <- seq(my.deming_err_Xonly$ci[1,1], my.deming_err_Xonly$ci[1,2], 0.0001)
deming.ribbon_table_Xonly <- data.frame("x" = x_range, "fit" = ((deming.slope_err_Xonly * x_range) + deming.int_err_Xonly), "y.min" = numeric(length = length(x_range)), "y.max" = numeric(length=length(x_range)))
deming.95ci.lines_Xonly <- data.frame("intercept" = rep(deming.int_seq_Xonly, each = length(deming.slope_seq_Xonly)), "slope" = deming.slope_seq_Xonly)
deming.ribbon_table_Xonly[,c("y.min","y.max")] <- t(sapply(x_range, getMaxMin, mydeming_lines = deming.95ci.lines_Xonly))
myCor_pear_Xonly <- cor.test(myData_Xonly$x, myData_Xonly$y, method=c("pearson"))

#Y only calculations
my.deming_err_Yonly <- deming(formula = myData_Yonly$y ~ myData_Yonly$x, xstd = myData_Yonly$x_err, ystd = myData_Yonly$y_err) # This works...it just takes FOREVER!
deming.slope_err_Yonly <- my.deming_err_Yonly$coefficients[2]
deming.slope_seq_Yonly <- seq(my.deming_err_Yonly$ci[2,1], my.deming_err_Yonly$ci[2,2], 0.001)
deming.int_err_Yonly <- my.deming_err_Yonly$coefficients[1]
deming.int_seq_Yonly <- seq(my.deming_err_Yonly$ci[1,1], my.deming_err_Yonly$ci[1,2], 0.0001)
deming.ribbon_table_Yonly <- data.frame("x" = x_range, "fit" = ((deming.slope_err_Yonly * x_range) + deming.int_err_Yonly), "y.min" = numeric(length = length(x_range)), "y.max" = numeric(length=length(x_range)))
deming.95ci.lines_Yonly <- data.frame("intercept" = rep(deming.int_seq_Yonly, each = length(deming.slope_seq_Yonly)), "slope" = deming.slope_seq_Yonly)
deming.ribbon_table_Yonly[,c("y.min","y.max")] <- t(sapply(x_range, getMaxMin, mydeming_lines = deming.95ci.lines_Yonly))
myCor_pear_Yonly <- cor.test(myData_Yonly$x, myData_Yonly$y, method=c("pearson"))
  
  
pdf(file="XY_ZFXonly_ZFYonly_Autosomes.pdf", height = 3, width = 3)
  ggplot() +
    geom_hline(yintercept =0,col="darkgrey", size=0.25) +
    geom_vline(xintercept = 0,  col="darkgrey", size=0.25) +
    geom_point(data=myData_Xonly, mapping=aes(x=x, y=y),col="#D95F0290", stroke=0, size=1.5) +
    geom_point(data=myData_Yonly, mapping=aes(x=x, y=y),col="#6e02d990", stroke=0, size=1.5) + 
    geom_point(data=myData_XY, mapping=aes(x=x, y=y),col="#40404060", stroke=0, size=1.5) + 
    geom_ribbon(data = deming.ribbon_table_Xonly, mapping = aes(x=x, ymin =y.min, ymax=y.max), fill="#D95F0230") +
    geom_abline(slope = deming.slope_err_Xonly, intercept = deming.int_err_Xonly,  col="#D95F02", size=0.5) +
    geom_ribbon(data = deming.ribbon_table_Yonly, mapping = aes(x=x, ymin =y.min, ymax=y.max), fill="#6e02d930") +
        geom_abline(slope = deming.slope_err_Yonly, intercept = deming.int_err_Yonly,  col="#6e02d9", size=0.5) +

    geom_ribbon(data = deming.ribbon_table_XY, mapping = aes(x=x, ymin =y.min, ymax=y.max), fill="#00000030") +
    geom_abline(slope = deming.slope_err_XY, intercept = deming.int_err_XY,  col="black", size=0.5) +
    coord_cartesian(xlim = c(myData_min, myData_max), ylim =c(myData_min, myData_max) ) +

    # scale_x_continuous(limits = c(myData_min, myData_max), expand = c(0,0)) +
    # scale_y_continuous(limits = c(myData_min, myData_max), expand = c(0,0)) +
    annotate(
      geom = "text",
      x = 0.1, #top left
      y = -1.1,
      label = paste0('ZFX & ZFY: r = ', formatC(myCor_pear_XY$estimate ,digits = 2,format = "f"), ", P = ", formatC(myCor_pear_XY$p.value, digits=2,format = "e")), hjust = 0,
      vjust = 1,
      size=4,
      col="black"
    ) +
    annotate(
      geom = "text",
      x = 0.1, #top left
      y = -1.3,
      label = paste0('ZFX only: r = ', formatC(myCor_pear_Xonly$estimate ,digits = 2,format = "f"), ", P = ", formatC(myCor_pear_Xonly$p.value, digits=2,format = "e")), hjust = 0,
      vjust = 1,
      size=4,
      col="#ff9933"
    ) +
    annotate(
      geom = "text",
      x = 0.1, #top left
      y = -1.5,
      label = paste0('ZFY only: r = ', formatC(myCor_pear_Yonly$estimate ,digits = 2,format = "f"), ", P = ", formatC(myCor_pear_Yonly$p.value, digits=2,format = "e")), hjust = 0,
      vjust = 1,
      size=4,
      col="#9966ff"
    ) +
    theme_classic(base_size = 10) +
    theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      aspect.ratio=1, 
      legend.position="none"
    )  +  labs(
      x= "ZFX kd vs control 46,XY\nlog2 fold change",
      y= "ZFY kd vs control 46,XY\nlog2 fold change"
    )
  dev.off()
  
  my.p <- c(5.1e-69, 6e-246, 9.2e-27)
my.p.adjust <- p.adjust(p = my.p, method = "bonferroni")
```

#Determine whether the cumulative levels of ZFX+ZFY differ between samples
## Run Tximport in length-scaled mode - use for analyzing summed expression of genes of different lengths (like ZFX+ZFY).
```{r}
#Merge ZFX and ZFY into one gene and re-run 
# tx2gene_zfx_y <- tx2gene
# tx2gene_zfx_y[tx2gene_zfx_y$GENEID == "ZFY","GENEID"] <- "ZFX"

# txi_zfx_zfy_F_merged_len <- tximport(txi_zfx_zfy_F_files_skd, type = "kallisto", tx2gene = tx2gene_zfx_y, countsFromAbundance = "lengthScaledTPM")
# save(txi_zfx_zfy_F_merged_len,file="txi_zfx_zfy_all_F_merged_lengthScaled.rda")
load(file="txi_zfx_zfy_all_F_merged_lengthScaled.rda")

#Male Single knockdowns only
# txi_zfx_zfy_M_skd_merged_len <- tximport(txi_zfx_zfy_M_files_skd, type = "kallisto", tx2gene = tx2gene_zfx_y, countsFromAbundance = "lengthScaledTPM")
# save(txi_zfx_zfy_M_skd_merged_len,file="txi_zfx_zfy_skd_M_merged_lengthScaled.rda")
load(file="txi_zfx_zfy_skd_M_merged_lengthScaled.rda")

#Male Double kds only
# txi_zfx_zfy_M_dkd_merged_len <- tximport(txi_zfx_zfy_M_files_dkd, type = "kallisto", tx2gene = tx2gene_zfx_y, countsFromAbundance = "lengthScaledTPM")
# save(txi_zfx_zfy_M_dkd_merged_len,file="txi_zfx_zfy_dkd_M_merged_lengthScaled.rda")
load(file="txi_zfx_zfy_dkd_M_merged_lengthScaled.rda")
```

##Make gene expn plots total expn ZFX+ZFY
```{r}
library(data.table)
myOrange <- "#d95f02"
myPurple <- "#7570B3"

#Cumulative expression
#XX SKD
xx_id <- paste0(metadata_skd[metadata_skd$karyotype == "XX","karyotype"],"_" , metadata_skd[metadata_skd$karyotype == "XX","genotype"])
xx_zfx_expn <- data.frame("category" = xx_id, "TPM" = txi_zfx_zfy_F_merged_len$abundance["ZFX",])
                          
#XY SKDs
xy_skd_id <- paste0(metadata_skd[metadata_skd$karyotype == "XY","karyotype"],"_" , metadata_skd[metadata_skd$karyotype == "XY","genotype"],"_skd")
xy_skd_zfx_zfy_expn <- data.frame("category" = xy_skd_id, "TPM" = txi_zfx_zfy_M_skd_merged_len$abundance["ZFX",])

#XY DKDs                          
xy_dkd_id <- paste0(metadata_dkd[metadata_dkd$karyotype == "XY","karyotype"],"_" , metadata_dkd[metadata_dkd$karyotype == "XY","genotype"],"_dkd")
xy_dkd_zfx_zfy_expn <- data.frame("category" = xy_dkd_id, "TPM" = txi_zfx_zfy_M_dkd_merged_len$abundance["ZFX",])

all_data <- data.table(rbind(xx_zfx_expn, xy_skd_zfx_zfy_expn, xy_dkd_zfx_zfy_expn))
plot_data_stats <- all_data[,.(median = summary(TPM)[3], firstq=summary(TPM)[2], thirdq = summary(TPM)[5]), by=category]


#ZFX epn
xx_id <- paste0(metadata_skd[metadata_skd$karyotype == "XX","karyotype"],"_" , metadata_skd[metadata_skd$karyotype == "XX","genotype"])
xy_skd_id<- paste0(metadata_skd[metadata_skd$karyotype == "XY","karyotype"],"_" , metadata_skd[metadata_skd$karyotype == "XY","genotype"],"_skd")
xy_dkd_id <- paste0(metadata_dkd[metadata_dkd$karyotype == "XY","karyotype"],"_" , metadata_dkd[metadata_dkd$karyotype == "XY","genotype"],"_dkd")

all_data_zfx <- rbind(
  data.frame("category" = xx_id, "gene" = "ZFX" ,"TPM" = txi_zfx_zfy_F_skd$abundance["ZFX",], "id" = paste0(xx_id,"_ZFX"), 
             row.names = names(txi_zfx_zfy_F_skd$abundance["ZFX",])),
  data.frame("category" = xy_skd_id, "gene" = "ZFX", "TPM" = txi_zfx_zfy_M_skd$abundance["ZFX",],"id" = paste0(xy_skd_id,"_ZFX"), 
             row.names = names(txi_zfx_zfy_M_skd$abundance["ZFX",])),
  data.frame("category" = xy_dkd_id, "gene" = "ZFX", "TPM" = txi_zfx_zfy_M_dkd$abundance["ZFX",],"id" = paste0(xy_dkd_id,"_ZFX"), 
             row.names = names(txi_zfx_zfy_M_dkd$abundance["ZFX",]))
)

all_data_zfx <- data.table(all_data_zfx, keep.rownames = TRUE)
                          
plot_data_stats_zfx <- as.data.frame(all_data_zfx[,.(median = summary(TPM)[3], firstq=summary(TPM)[2], thirdq = summary(TPM)[5]), by=category])

pdf(file="ZFX_Y_TPM_CRISPRi_cumulative.pdf", width=2, height=1.75)
ggplot(data = plot_data_stats, mapping = aes(x = category, y = median)) +
geom_bar(stat="identity", fill=myPurple) +
geom_bar(data = plot_data_stats_zfx, mapping = aes(x=category, y = median), stat="identity", fill= myOrange) +
geom_errorbar(data=plot_data_stats,aes(ymin = firstq, ymax = thirdq), width = 0.2) +
theme_classic(base_size = 8) + 
	theme(
	axis.text = element_text(color = "black"), 
	axis.ticks = element_line(color = "black", size = 0.25), 
	axis.line = element_line(size = 0.25),
	plot.title = element_text(hjust = 0.5),
	axis.text.x = element_text(hjust = 0.5),
	legend.position = "none",
	) + ylab("TPM") + ylim(0,26)
dev.off()


```

##Calculate difference in single vs double kd
```{r}
xx_zfx_expn_cntrl_mean <- mean(xx_zfx_expn[xx_zfx_expn$category == "XX_Control","TPM"])
xx_zfx_expn_update <- xx_zfx_expn
xx_zfx_expn_update$FoldChange <- xx_zfx_expn$TPM / xx_zfx_expn_cntrl_mean 

xy_skd_zfx_zfy_expn_cntrl_mean <- mean(xy_skd_zfx_zfy_expn[xy_skd_zfx_zfy_expn$category == "XY_Control_skd","TPM"])
xy_skd_zfx_zfy_expn_update <- xy_skd_zfx_zfy_expn
xy_skd_zfx_zfy_expn_update$FoldChange <- xy_skd_zfx_zfy_expn$TPM / xy_skd_zfx_zfy_expn_cntrl_mean 

xy_dkd_zfx_zfy_expn_cntrl_mean <- mean(xy_dkd_zfx_zfy_expn[xy_dkd_zfx_zfy_expn$category == "XY_Control_dkd","TPM"])
xy_dkd_zfx_zfy_expn_update <- xy_dkd_zfx_zfy_expn
xy_dkd_zfx_zfy_expn_update$FoldChange <- xy_dkd_zfx_zfy_expn$TPM / xy_dkd_zfx_zfy_expn_cntrl_mean

#compare single kds to double kd fold-changes in cumulative expression!
#XX zfx vs dkd
wilcox.test(x = xx_zfx_expn_update[xx_zfx_expn_update$category == "XX_ZFX_kd","FoldChange"],
            y=xy_dkd_zfx_zfy_expn_update[xy_dkd_zfx_zfy_expn_update$category == "XY_ZFX_ZFY_kd_dkd","FoldChange"])

#XY zfx vs dkd
wilcox.test(x = xy_skd_zfx_zfy_expn_update[xy_skd_zfx_zfy_expn_update$category == "XY_ZFX_kd_skd","FoldChange"], y=xy_dkd_zfx_zfy_expn_update[xy_dkd_zfx_zfy_expn_update$category == "XY_ZFX_ZFY_kd_dkd","FoldChange"])

#XY zfy vs dkd
wilcox.test(x = xy_skd_zfx_zfy_expn_update[xy_skd_zfx_zfy_expn_update$category == "XY_ZFY_kd_skd","FoldChange"], y=xy_dkd_zfx_zfy_expn_update[xy_dkd_zfx_zfy_expn_update$category == "XY_ZFX_ZFY_kd_dkd","FoldChange"])

#XY zfx vs zfy
wilcox.test(x = xy_skd_zfx_zfy_expn_update[xy_skd_zfx_zfy_expn_update$category == "XY_ZFX_kd_skd","FoldChange"], 
            y= xy_skd_zfx_zfy_expn_update[xy_skd_zfx_zfy_expn_update$category == "XY_ZFY_kd_skd","FoldChange"])

```


#Compare knockdowns to X and Y response
## Venn diagrams
```{r}
#Union of all ZFX or ZFY responsive genes:
union_all_zfx_zfy <- union(union(union(male_skd_res_zfx_cntrl_sig_auto[,"gene_name.107"], male_skd_res_zfx_cntrl_sig_auto[,"gene_name.107"]),female_res_zfx_cntrl_sig_auto[,"gene_name.107"]), male_dkd_res_dko_cntrl_auto_sig[,"gene_name.107"])


#Compare X responsive, Y responsive, union zfx_y responsive
threeGroupVenn_geneNames(setA = fib_results_sig_x[,"Gene"], setAname = "X_responsive", setB =fib_results_sig_y[,"Gene"], setBname = "Y_responsive", setC = union_all_zfx_zfy, setCname = "Union_ZFX_ZFY_responsive", expGenes = fib_expressed$expAutoGenes, filename = "X_resp_Y_resp_CRISPRi_union.pdf")

twoGroupVenn_geneNames(setA = fib_results_sig_x_y[,"Gene"], setAname = "X&Y_responsive", setB = union_all_zfx_zfy, setBname = "Union_ZFX_ZFY_responsive", expGenes = fib_expressed$expAutoGenes, filename = "ZFX_union_vs_X&Y_resp_VENN.pdf")

```

##X only correlation plots!
```{r}
#### X only genes! ####
myData <- na.omit(data.frame("x" = female_res_zfx_cntrl_exp[fib_results_sig_x_only$gene_name.84,"log2FoldChange"], "x_err" = female_res_zfx_cntrl_exp[fib_results_sig_x_only$gene_name.84,"lfcSE"],  "y" = fib_results_sig_x_only[,"fib_log2FC_X"], "y_err" = fib_results_sig_x_only[,"fib_lfcSE_X"], row.names = fib_results_sig_x_only$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "X_only_genes_XX_ZFX_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFXkd vs control 46,XX\nlog2 fold change", myYlab = "Log2 fold change per Chr X",  negIdentity = TRUE,  myYlim = c(-3,3), myXlim=c(-3,3), printRsq = TRUE, pointsCol = orange_pointsCol, demingBehind = TRUE)

#Do XY ZFX kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfx_cntrl_exp[fib_results_sig_x_only$gene_name.84,"log2FoldChange"],"x_err" = male_skd_res_zfx_cntrl_exp[fib_results_sig_x_only$gene_name.84,"lfcSE"], "y" = fib_results_sig_x_only[,"fib_log2FC_X"], 
                             "y_err" = fib_results_sig_x_only[,"fib_lfcSE_X"], row.names = (fib_results_sig_x_only$gene_name.84)))

paper2_corrPlot_deming_err(myData = myData, myTitle = "X_only_genes_XY_ZFX_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFXkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr X",negIdentity = TRUE, printRsq = TRUE, pointsCol = orange_pointsCol, demingBehind = TRUE)


#Do XY ZFY kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfy_cntrl_exp[fib_results_sig_x_only$gene_name.84,"log2FoldChange"], "x_err" = male_skd_res_zfy_cntrl_exp[fib_results_sig_x_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_x_only[,"fib_log2FC_X"], "y_err" = fib_results_sig_x_only[,"fib_lfcSE_X"], row.names = (fib_results_sig_x_only$gene_name.84)))

paper2_corrPlot_deming_err(myData = myData, myTitle = "X_only_genes_XY_ZFY_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr X", negIdentity = TRUE,  printRsq = TRUE, pointsCol = orange_pointsCol, demingBehind = TRUE)

#Do XY dkd now
myData <- na.omit(data.frame("x" = male_dkd_res_dko_cntrl_exp[fib_results_sig_x_only$gene_name.84,"log2FoldChange"], "x_err" = male_dkd_res_dko_cntrl_exp[fib_results_sig_x_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_x_only[,"fib_log2FC_X"], "y_err" = fib_results_sig_x_only[,"fib_lfcSE_X"],row.names = (fib_results_sig_x_only$gene_name.84)))

paper2_corrPlot_deming_err(myData = myData, myTitle = "X_only_genes_XY_ZFXZFY_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFX_ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr X", negIdentity = TRUE, printRsq = TRUE, pointsCol = orange_pointsCol, demingBehind = TRUE)
```

##Y only correlation plots!
```{r}
myData <- na.omit(data.frame("x" = female_res_zfx_cntrl_exp[fib_results_sig_y_only$gene_name.84,"log2FoldChange"], "x_err" = female_res_zfx_cntrl_exp[fib_results_sig_y_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_y_only[,"fib_log2FC_Y"], "y_err" = fib_results_sig_y_only[,"fib_lfcSE_Y"],  row.names = fib_results_sig_y_only$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "Y_only_genes_XX_ZFX_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFXkd vs control 46,XX\nlog2 fold change", myYlab = "Log2 fold change per Chr Y",  negIdentity = TRUE, printRsq = TRUE, pointsCol = purple_pointsCol, demingBehind = TRUE)

#Do XY ZFX kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfx_cntrl_exp[fib_results_sig_y_only$gene_name.84,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[fib_results_sig_y_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_y_only[,"fib_log2FC_Y"],  "y_err" = fib_results_sig_y_only[,"fib_lfcSE_Y"], row.names = fib_results_sig_y_only$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "Y_only_genes_XY_ZFX_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFXkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr Y",negIdentity = TRUE, printRsq = TRUE, pointsCol = purple_pointsCol, demingBehind = TRUE)


#Do XY ZFY kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfy_cntrl_exp[fib_results_sig_y_only$gene_name.84,"log2FoldChange"], "x_err" = male_skd_res_zfy_cntrl_exp[fib_results_sig_y_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_y_only[,"fib_log2FC_Y"], "y_err" = fib_results_sig_y_only[,"fib_lfcSE_Y"], row.names = fib_results_sig_y_only$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "Y_only_genes_XY_ZFY_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr Y", negIdentity = TRUE, printRsq = TRUE, pointsCol = purple_pointsCol, demingBehind = TRUE)

#Do XY dkd now
myData <- na.omit(data.frame("x" = male_dkd_res_dko_cntrl_exp[fib_results_sig_y_only$gene_name.84,"log2FoldChange"], "x_err" = male_dkd_res_dko_cntrl_exp[fib_results_sig_y_only$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_y_only[,"fib_log2FC_Y"],  "y_err" = fib_results_sig_y_only[,"fib_lfcSE_Y"],row.names = fib_results_sig_y_only$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "Y_only_genes_XY_ZFXZFY_vs_Xresp.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFX_ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Log2 fold change per Chr Y", negIdentity = TRUE, printRsq = TRUE, pointsCol = purple_pointsCol, demingBehind = TRUE)
```

##X and Y correlation plots!
```{r}
myData <- na.omit(data.frame("x" = female_res_zfx_cntrl_exp[fib_results_sig_x_y$gene_name.84,"log2FoldChange"], "x_err" = female_res_zfx_cntrl_exp[fib_results_sig_x_y$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_x_y[,"fib_log2FC_XY_avg"], "y_err" = fib_results_sig_x_y[,"fib_lfcSE_XY_avg"],
                             row.names = fib_results_sig_x_y$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "XY_genes_XX_ZFX_vs_Xresp.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFXkd vs control 46,XX\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y",  negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)

#Do XY ZFX kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfx_cntrl_exp[fib_results_sig_x_y$gene_name.84,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[fib_results_sig_x_y$gene_name.84,"lfcSE"], 
                             "y" = fib_results_sig_x_y[,"fib_log2FC_XY_avg"], "y_err" = fib_results_sig_x_y[,"fib_lfcSE_XY_avg"], 
                             row.names = fib_results_sig_x_y$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "XY_genes_XY_ZFX_vs_Xresp.pdf",  myWidth = 1.6, myHeight = 1.6, myXlab = "ZFXkd vs control 46,XY\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y",negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)


#Do XY ZFY kd now
myData <- na.omit(data.frame("x" = male_skd_res_zfy_cntrl_exp[fib_results_sig_x_y$gene_name.84,"log2FoldChange"], "x_err" = male_skd_res_zfy_cntrl_exp[fib_results_sig_x_y$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_x_y[,"fib_log2FC_XY_avg"],  "y_err" = fib_results_sig_x_y[,"fib_lfcSE_XY_avg"], row.names = fib_results_sig_x_y$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "XY_genes_XY_ZFY_vs_Xresp.pdf",  myWidth = 1.65, myHeight = 1.65, myXlab = "ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y", negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)

#Do XY dkd now
myData <- na.omit(data.frame("x" = male_dkd_res_dko_cntrl_exp[fib_results_sig_x_y$gene_name.84,"log2FoldChange"],"x_err" = male_dkd_res_dko_cntrl_exp[fib_results_sig_x_y$gene_name.84,"lfcSE"],
                             "y" = fib_results_sig_x_y[,"fib_log2FC_XY_avg"],  "y_err" = fib_results_sig_x_y[,"fib_lfcSE_XY_avg"],
                             row.names = fib_results_sig_x_y$gene_name.84))

paper2_corrPlot_deming_err(myData = myData, myTitle = "XY_genes_XY_ZFXZFY_vs_Xresp.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFX_ZFYkd vs control 46,XY\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y", negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)
```

#Show what happens for the Xa Fib genes that are X and Y responsive
```{r}
#bring AR annotations
AR_values <- read.delim(file=paste0(myPath,"Annotations/all_AR_annotation_20230530.txt"))
rownames(AR_values) <- AR_values$gene_name.84

AR_values_subject <- AR_values[AR_values$Final_XCI_Call %in% c("Subject"),]

#bring in log2FC fibs
fib_results_X_ALL_genes <- read.delim(file=paste0(myPath,"Linear_regressions_new/Fibroblasts/fib_x_count_expressed.txt"), stringsAsFactors = FALSE, row.names = 1)
fib_results_Y_ALL_genes <- read.delim(file=paste0(myPath,"Linear_regressions_new/Fibroblasts/fib_y_count_expressed.txt"), stringsAsFactors = FALSE)

fib_res_X_subject <- fib_results_X_ALL_genes[rownames(fib_results_X_ALL_genes) %in% rownames(AR_values_subject),] #315 genes
fib_res_Y_subject <- fib_results_Y_ALL_genes[rownames(fib_results_Y_ALL_genes) %in% rownames(AR_values_subject),] #315 genes

fib_subject_sig_X <- fib_res_X_subject[fib_res_X_subject$padj < 0.05,]
fib_subject_sig_Y <- fib_res_Y_subject[fib_res_Y_subject$padj < 0.05,]

fib_subject_sig_X_Y_genes <- intersect(fib_subject_sig_X$Gene, fib_subject_sig_Y$Gene)

fib_subject_sig_X_and_Y <- data.frame("avg_log2FC" = (fib_subject_sig_X[fib_subject_sig_X_Y_genes,"log2FoldChange"] + fib_subject_sig_Y[fib_subject_sig_X_Y_genes,"log2FoldChange"] / 2), row.names = fib_subject_sig_X_Y_genes)


#How many of Xa genes are ZFX and ZFY responsive?
male_skd_zfx_Xaonly_sig <- male_skd_res_zfx_cntrl_exp[male_skd_res_zfx_cntrl_exp$Gene %in% fib_res_X_subject$Gene & male_skd_res_zfx_cntrl_exp$padj < 0.05,] #29

male_skd_zfy_Xaonly_sig <- male_skd_res_zfy_cntrl_exp[male_skd_res_zfy_cntrl_exp$Gene %in% fib_res_X_subject$Gene & male_skd_res_zfy_cntrl_exp$padj < 0.05,] #6

male_dkd_zfx_zfy_Xaonly_sig <- male_dkd_res_dko_cntrl_exp[male_dkd_res_dko_cntrl_exp$Gene %in% fib_res_X_subject$Gene & male_dkd_res_dko_cntrl_exp$padj < 0.05,] #45

female_skd_zfx_Xaonly_sig <- female_res_zfx_cntrl_exp[female_res_zfx_cntrl_exp$Gene %in% fib_res_X_subject$Gene & female_res_zfx_cntrl_exp$padj < 0.05,] #128

union_all_zfx_zfy <- union(male_skd_zfx_Xaonly_sig$Gene, union(male_skd_zfy_Xaonly_sig$Gene, union(male_dkd_zfx_zfy_Xaonly_sig$Gene, female_skd_zfx_Xaonly_sig$Gene))) #138

threeGroupVenn_geneNames(setA=female_skd_zfx_Xaonly_sig$Gene, setAname = "XX_zfx_kd", setB=male_skd_zfx_Xaonly_sig$Gene, setBname = "XY_zfx_kd", setC=male_skd_zfy_Xaonly_sig$Gene, setCname = "XY_ZFY_kd", expGenes = fib_res_X_subject$Gene, filename = "Xa_only_CRISPRi_sig_venn.pdf")

#Plot the genes that respond to both ZFX and ZFY
myData <- data.frame("y" = male_skd_zfy_Xaonly_sig[,"log2FoldChange"], "x" = male_skd_zfx_Xaonly_sig[male_skd_zfy_Xaonly_sig$Gene, "log2FoldChange"], row.names = male_skd_zfy_Xaonly_sig$Gene)

paper2_corrPlot_deming_err(myData = myData, myTitle = "ZFX_ZFY_responsive gene_XY.pdf", myWidth = 1.75, myHeight = 1.75, myXlab = "ZFXkd vs control 46,XY\nlog2 fold-change", myYlab = "ZFYkd vs control 46,XY\nlog2 fold-change", demingBehind = TRUE)

#Venn diagram with sig X+Y responsive genes
twoGroupVenn_geneNames(setA = union_all_zfx_zfy, setAname = "ZFX & ZFY union", setB = fib_subject_sig_X_Y_genes, setBname = "Sig X and Y Xa",  expGenes = fib_res_X_subject$Gene, filename = "ZFX_Y_union_vs_Xaexpressed_XYsig.pdf")

# Plot
myData <- data.frame("y" = fib_subject_sig_X_and_Y$avg_log2FC, "x" = female_res_zfx_cntrl_exp[rownames(fib_subject_sig_X_and_Y), "log2FoldChange"], row.names = rownames(fib_subject_sig_X_and_Y))

myData <- data.frame("y" = ((fib_subject_sig_Y[fib_subject_sig_X_Y_genes,"log2FoldChange"] +
                                fib_subject_sig_X[fib_subject_sig_X_Y_genes,"log2FoldChange"])/2), 
                     "x" = female_res_zfx_cntrl_exp[fib_subject_sig_X_Y_genes, "log2FoldChange"], 
                     row.names = fib_subject_sig_X_Y_genes)

paper2_corrPlot_deming(myData = myData, myTitle = "corrPlot_fib_Xa_X_Y_resp_vs_46XX_cntrlvsZFXkd.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFX kd vs control 46,XX\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y",  negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)

#XY
myData <- data.frame("y" = ((fib_subject_sig_Y[fib_subject_sig_X_Y_genes,"log2FoldChange"] +
                                fib_subject_sig_X[fib_subject_sig_X_Y_genes,"log2FoldChange"])/2), 
                     "x" = male_skd_res_zfx_cntrl_exp[fib_subject_sig_X_Y_genes, "log2FoldChange"], 
                     row.names = fib_subject_sig_X_Y_genes)
paper2_corrPlot_deming(myData = myData, myTitle = "corrPlot_fib_Xa_X_Y_resp_vs_46XY_cntrlvsZFXkd.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFX kd vs control 46,XY\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y",   negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)


myData <- data.frame("y" = ((fib_subject_sig_Y[fib_subject_sig_X_Y_genes,"log2FoldChange"] +
                                fib_subject_sig_X[fib_subject_sig_X_Y_genes,"log2FoldChange"])/2), 
                     "x" = male_skd_res_zfy_cntrl_exp[fib_subject_sig_X_Y_genes, "log2FoldChange"], 
                     row.names = fib_subject_sig_X_Y_genes)

paper2_corrPlot_deming(myData = myData, myTitle = "corrPlot_fib_Xa_X_Y_resp_vs_46XY_cntrlvsZFYkd.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFY kd vs control 46,XY\nlog2 fold change", myYlab = "Average log2FC\nper Chr X & Y",  negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)


myData <- data.frame("y" = ((fib_subject_sig_Y[fib_subject_sig_X_Y_genes,"log2FoldChange"] +
                                fib_subject_sig_X[fib_subject_sig_X_Y_genes,"log2FoldChange"])/2), 
                     "x" = male_dkd_res_dko_cntrl_exp[fib_subject_sig_X_Y_genes, "log2FoldChange"], 
                     row.names = fib_subject_sig_X_Y_genes)
paper2_corrPlot_deming(myData = myData, myTitle = "corrPlot_fib_Xa_X_Y_resp_vs_46XY_cntrlvsdkd.pdf", myWidth = 1.6, myHeight = 1.6, myXlab = "ZFX & ZFY kd vs control 46,XY\nlog2 fold change",myYlab = "Average log2FC\nper Chr X & Y",   negIdentity = TRUE, printRsq = TRUE, pointsCol = default_pointsCol, demingBehind = TRUE)



##2 plots

male_skd_zfx_Xaonly_sig <- male_skd_res_zfx_cntrl_exp[male_skd_res_zfx_cntrl_exp$Gene %in% fib_res_X_subject$Gene & male_skd_res_zfx_cntrl_exp$padj < 0.05,] #29

male_skd_zfy_Xaonly_sig <- male_skd_res_zfy_cntrl_exp[male_skd_res_zfy_cntrl_exp$Gene %in% fib_res_X_subject$Gene & male_skd_res_zfy_cntrl_exp$padj < 0.05,] #6

zfx_resp_genes_XY_Xa <- setdiff(male_skd_zfx_Xaonly_sig$Gene, male_skd_zfy_Xaonly_sig$Gene)

myData_XY <- data.frame("x" = male_skd_res_zfx_cntrl_exp[male_skd_zfy_Xaonly_sig$Gene,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[male_skd_zfy_Xaonly_sig$Gene,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[male_skd_zfy_Xaonly_sig$Gene,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[male_skd_zfy_Xaonly_sig$Gene,"lfcSE"])

myData_Xonly <- data.frame("x" = male_skd_res_zfx_cntrl_exp[zfx_resp_genes_XY_Xa,"log2FoldChange"], "y" = male_skd_res_zfy_cntrl_exp[zfx_resp_genes_XY_Xa,"log2FoldChange"], "x_err" = male_skd_res_zfx_cntrl_exp[zfx_resp_genes_XY_Xa,"lfcSE"],
                     "y_err" = male_skd_res_zfy_cntrl_exp[zfx_resp_genes_XY_Xa,"lfcSE"])

myLim <- makeSymmetric_X_Y(myData_XY$x, myData_XY$y)
# myLim_Xonly <- makeSymmetric_X_Y(myData_Xonly$x, myData_Xonly$y)
# myLim_Yonly <- makeSymmetric_X_Y(myData_Yonly$x, myData_Yonly$y)

myData_min <- myLim[1] - 0.1
myData_max <- myLim[2] + 0.1
x_range <- seq(myData_min - 1, myData_max + 1, 0.01)

#X and Y calculations
my.deming_err_XY <- deming(formula = myData_XY$y ~ myData_XY$x, xstd = myData_XY$x_err, ystd = myData_XY$y_err) # This works...it just takes FOREVER!
deming.slope_err_XY <- my.deming_err_XY$coefficients[2]
deming.slope_seq_XY <- seq(my.deming_err_XY$ci[2,1], my.deming_err_XY$ci[2,2], 0.001)
deming.int_err_XY <- my.deming_err_XY$coefficients[1]
deming.int_seq_XY <- seq(my.deming_err_XY$ci[1,1], my.deming_err_XY$ci[1,2], 0.0001)
deming.ribbon_table_XY <- data.frame("x" = x_range, "fit" = ((deming.slope_err_XY * x_range) + deming.int_err_XY), "y.min" = numeric(length = length(x_range)), "y.max" = numeric(length=length(x_range)))
deming.95ci.lines_XY <- data.frame("intercept" = rep(deming.int_seq_XY, each = length(deming.slope_seq_XY)), "slope" = deming.slope_seq_XY)
deming.ribbon_table_XY[,c("y.min","y.max")] <- t(sapply(x_range, getMaxMin, mydeming_lines = deming.95ci.lines_XY))
myCor_pear_XY <- cor.test(myData_XY$x, myData_XY$y, method=c("pearson"))


#X only calculations
my.deming_err_Xonly <- deming(formula = myData_Xonly$y ~ myData_Xonly$x, xstd = myData_Xonly$x_err, ystd = myData_Xonly$y_err) # This works...it just takes FOREVER!
deming.slope_err_Xonly <- my.deming_err_Xonly$coefficients[2]
deming.slope_seq_Xonly <- seq(my.deming_err_Xonly$ci[2,1], my.deming_err_Xonly$ci[2,2], 0.001)
deming.int_err_Xonly <- my.deming_err_Xonly$coefficients[1]
deming.int_seq_Xonly <- seq(my.deming_err_Xonly$ci[1,1], my.deming_err_Xonly$ci[1,2], 0.0001)
deming.ribbon_table_Xonly <- data.frame("x" = x_range, "fit" = ((deming.slope_err_Xonly * x_range) + deming.int_err_Xonly), "y.min" = numeric(length = length(x_range)), "y.max" = numeric(length=length(x_range)))
deming.95ci.lines_Xonly <- data.frame("intercept" = rep(deming.int_seq_Xonly, each = length(deming.slope_seq_Xonly)), "slope" = deming.slope_seq_Xonly)
deming.ribbon_table_Xonly[,c("y.min","y.max")] <- t(sapply(x_range, getMaxMin, mydeming_lines = deming.95ci.lines_Xonly))
myCor_pear_Xonly <- cor.test(myData_Xonly$x, myData_Xonly$y, method=c("pearson"))

pdf(file="XY_ZFXonly_Xa.pdf", height = 2, width = 2)
  ggplot() +
    geom_hline(yintercept =0,col="darkgrey", size=0.25) +
    geom_vline(xintercept = 0,  col="darkgrey", size=0.25) +
    geom_point(data=myData_Xonly, mapping=aes(x=x, y=y),col="#D95F0290", stroke=0) +
    geom_point(data=myData_XY, mapping=aes(x=x, y=y),col="#40404060", stroke=0) + 
    geom_ribbon(data = deming.ribbon_table_Xonly, mapping = aes(x=x, ymin =y.min, ymax=y.max), fill="#D95F0230") +
    geom_abline(slope = deming.slope_err_Xonly, intercept = deming.int_err_Xonly,  col="#D95F02", size=0.5) +
    geom_ribbon(data = deming.ribbon_table_XY, mapping = aes(x=x, ymin =y.min, ymax=y.max), fill="#00000030") +
    geom_abline(slope = deming.slope_err_XY, intercept = deming.int_err_XY,  col="black", size=0.5) +
    coord_cartesian(xlim = c(myData_min, myData_max), ylim =c(myData_min, myData_max) ) +

    # scale_x_continuous(limits = c(myData_min, myData_max), expand = c(0,0)) +
    # scale_y_continuous(limits = c(myData_min, myData_max), expand = c(0,0)) +
    annotate(
      geom = "text",
      x = 0.1, #top left
      y = -0.7,
      label = paste0('X & Y: r = ', formatC(myCor_pear_XY$estimate ,digits = 2,format = "f"), ", P = ", formatC(myCor_pear_XY$p.value, digits=2,format = "e")), hjust = 0,
      vjust = 1,
      size=2,
      col="black"
    ) +
    annotate(
      geom = "text",
      x = 0.1, #top left
      y = -0.9,
      label = paste0('X only: r = ', formatC(myCor_pear_Xonly$estimate ,digits = 2,format = "f"), ", P = ", formatC(myCor_pear_Xonly$p.value, digits=2,format = "e")), hjust = 0,
      vjust = 1,
      size=2,
      col="#ff9933"
    ) +
    theme_classic(base_size = 8) +
    theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      aspect.ratio=1, 
      legend.position="none"
    )  +  labs(
      x= "ZFX kd vs control 46,XY\nlog2 fold change",
      y= "ZFY kd vs control 46,XY\nlog2 fold change"
    )
  dev.off()
  
  my.p <- c(8.09e-4, 4.7e-9)
my.p.adjust <- p.adjust(p = my.p, method = "bonferroni")
my.p.adjust
```

#Compare to previous ZFX/ZFY publications
```{r}
#bring in direct targets 
zfx_targets_mcf7 <- read.delim(file="ZFX_ChIPseq/mcf7_direct_targets.txt",  stringsAsFactors = FALSE)[,c(8,6,2)]
rownames(zfx_targets_mcf7) <- zfx_targets_mcf7$Gene
colnames(zfx_targets_mcf7) <- c("Gene","MCF7_padj","MCF7_log2FC")

zfx_targets_C42B <- read.delim(file="ZFX_ChIPseq/c42b_direct_targets.txt",  stringsAsFactors = FALSE)[,c(8,6,2)]
rownames(zfx_targets_C42B) <- zfx_targets_C42B$Gene
colnames(zfx_targets_C42B) <- c("Gene","C42B_padj","C42B_log2FC")

zfx_targets_HEK <- read.delim(file="ZFX_ChIPseq/hek_direct_targets.txt",  stringsAsFactors = FALSE)[,c(8,6,2)]
rownames(zfx_targets_HEK) <- zfx_targets_HEK$Gene

colnames(zfx_targets_HEK) <- c("Gene","hek_padj","hek_log2FC")

#For stats, need to know what genes are expressed in these tissues
mcf7_expressed <- read.delim(file="ZFX_ChIPseq/mcf7_expressed.txt", stringsAsFactors = FALSE)
c42b_expressed <- read.delim(file="ZFX_ChIPseq/c42b_expressed.txt", stringsAsFactors = FALSE)
hek_expressed <- read.delim(file="ZFX_ChIPseq/hek293t_expressed.txt", stringsAsFactors = FALSE)

#Make barplots for the results
#Merge results
lcl_fib_mcf7 <- merge(x=lcl_fib_results, y=zfx_targets_mcf7, by.x="gene_name.84", by.y="Gene", all.x=TRUE)
lcl_fib_mcf7_c42b <- merge(x=lcl_fib_mcf7, y=zfx_targets_C42B, by.x="gene_name.84", by.y="Gene", all.x=TRUE)
lcl_fib_mcf7_c42b_hek <-  merge(x=lcl_fib_mcf7_c42b, y=zfx_targets_HEK, by.x="gene_name.84", by.y="Gene", all.x=TRUE)
lcl_fib_mcf7_c42b_hek$mcf7_expressed <- "No"
lcl_fib_mcf7_c42b_hek$c42b_expressed <- "No"
lcl_fib_mcf7_c42b_hek$hek_expressed <- "No"
lcl_fib_mcf7_c42b_hek[lcl_fib_mcf7_c42b_hek$Gene %in% mcf7_expressed$gene_name.107, "mcf7_expressed"] <- "Yes"
lcl_fib_mcf7_c42b_hek[lcl_fib_mcf7_c42b_hek$Gene %in% c42b_expressed$gene_name.107, "c42b_expressed"] <- "Yes"
lcl_fib_mcf7_c42b_hek[lcl_fib_mcf7_c42b_hek$Gene %in% hek_expressed$gene_name.107, "hek_expressed"] <- "Yes"

lcl_results <- lcl_fib_mcf7_c42b_hek[! is.na(lcl_fib_mcf7_c42b_hek$lcl_padj_X), ]
lcl_results_x_sig_up <- lcl_results[lcl_results$lcl_padj_X < 0.05 & lcl_results$lcl_log2FC_X > 0,]
lcl_results_x_sig_dn <- lcl_results[lcl_results$lcl_padj_X < 0.05 & lcl_results$lcl_log2FC_X < 0,]
lcl_results_y_sig_up <- lcl_results[lcl_results$lcl_padj_Y < 0.05 & lcl_results$lcl_log2FC_Y > 0,]
lcl_results_y_sig_dn <- lcl_results[lcl_results$lcl_padj_Y < 0.05 & lcl_results$lcl_log2FC_Y < 0,]
lcl_results_xy_sig_up <- lcl_results[lcl_results$lcl_padj_X < 0.05 & lcl_results$lcl_log2FC_X > 0 & lcl_results$lcl_padj_Y < 0.05 & lcl_results$lcl_log2FC_Y > 0,]
lcl_results_xy_sig_dn <- lcl_results[lcl_results$lcl_padj_X < 0.05 & lcl_results$lcl_log2FC_X < 0 & lcl_results$lcl_padj_Y < 0.05 & lcl_results$lcl_log2FC_Y < 0,]

public_data <- c("mcf7_up", "mcf7_dn", "c42b_up","c42b_dn","hek_up","hek_dn")
myCols <- c("#FA807250", "#FA8072", "#20B2AA50","#20B2AA","#BA55D350", "#BA55D3")

total_lcl_mcf7_exp <- dim(lcl_results[lcl_results$mcf7_expressed == "Yes",])[1]
total_lcl_c42b_exp <- dim(lcl_results[lcl_results$c42b_expressed == "Yes",])[1]
total_lcl_hek_exp <- dim(lcl_results[lcl_results$hek_expressed == "Yes",])[1]

myData <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())

#Up
myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$MCF7_padj) & lcl_results_x_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$mcf7_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$MCF7_padj) &
                                                                                            lcl_results_x_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$MCF7_padj) & lcl_results_x_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$mcf7_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$MCF7_padj) & lcl_results_x_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$C42B_padj) & lcl_results_x_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$c42b_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$C42B_padj) & lcl_results_x_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$C42B_padj) & lcl_results_x_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$C42B_padj) & lcl_results_x_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$hek_padj) & lcl_results_x_sig_up$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$hek_padj) & lcl_results_x_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$hek_padj) & lcl_results_x_sig_up$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_up[lcl_results_x_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_up[! is.na(lcl_results_x_sig_up$hek_padj) & lcl_results_x_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_up[lcl_results_x_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$MCF7_padj) & lcl_results_x_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$MCF7_padj) & lcl_results_x_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$MCF7_padj) & lcl_results_x_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$MCF7_padj) & lcl_results_x_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$C42B_padj) & lcl_results_x_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$C42B_padj) & lcl_results_x_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$C42B_padj) & lcl_results_x_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$C42B_padj) & lcl_results_x_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$hek_padj) & lcl_results_x_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$hek_padj) & lcl_results_x_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$hek_padj) & lcl_results_x_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_x_sig_dn[! is.na(lcl_results_x_sig_dn$hek_padj) & lcl_results_x_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_x_sig_dn[lcl_results_x_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_X <- myData
myData_plot_X$Category <- factor(x=myData_plot_X$Category, levels = c("X_up","X_dn"))
myData_plot_X$Public_data <- factor(x=myData_plot_X$Public_data, levels = public_data)
myData_plot_X[myData_plot_X$Percentage == 0, "Percentage"] <- 0.1

pdf(file="X_resp_LCL_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_X, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()


#Y response
myData_Y <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())
#No change

#Up
myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$MCF7_padj) & lcl_results_y_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$MCF7_padj) & lcl_results_y_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$MCF7_padj) & lcl_results_y_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$MCF7_padj) & lcl_results_y_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$C42B_padj) & lcl_results_y_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$C42B_padj) & lcl_results_y_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$C42B_padj) & lcl_results_y_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$C42B_padj) & lcl_results_y_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$hek_padj) & lcl_results_y_sig_up$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$hek_padj) & lcl_results_y_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$hek_padj) & lcl_results_y_sig_up$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_up[lcl_results_y_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_up[! is.na(lcl_results_y_sig_up$hek_padj) & lcl_results_y_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_up[lcl_results_y_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$MCF7_padj) & lcl_results_y_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$MCF7_padj) & lcl_results_y_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$MCF7_padj) & lcl_results_y_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$MCF7_padj) & lcl_results_y_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$C42B_padj) & lcl_results_y_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$C42B_padj) & lcl_results_y_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$C42B_padj) & lcl_results_y_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$C42B_padj) & lcl_results_y_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$hek_padj) & lcl_results_y_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$hek_padj) & lcl_results_y_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$hek_padj) & lcl_results_y_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_y_sig_dn[! is.na(lcl_results_y_sig_dn$hek_padj) & lcl_results_y_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_y_sig_dn[lcl_results_y_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_Y <- myData_Y
myData_plot_Y$Category <- factor(x=myData_plot_Y$Category, levels = c("Y_up","Y_dn"))
myData_plot_Y$Public_data <- factor(x=myData_plot_Y$Public_data, levels = public_data)
myData_plot_Y[myData_plot_Y$Percentage == 0, "Percentage"] <- 0.1

pdf(file="Y_resp_LCL_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_Y, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()


#X and Y response
myData_XY <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())
#No change

#Up
myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$MCF7_padj) & lcl_results_xy_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$MCF7_padj) & lcl_results_xy_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$MCF7_padj) & lcl_results_xy_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$MCF7_padj) & lcl_results_xy_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$C42B_padj) & lcl_results_xy_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$C42B_padj) & lcl_results_xy_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$C42B_padj) & lcl_results_xy_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$C42B_padj) & lcl_results_xy_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$hek_padj) & lcl_results_xy_sig_up$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$hek_padj) & lcl_results_xy_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$hek_padj) & lcl_results_xy_sig_up$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_up[! is.na(lcl_results_xy_sig_up$hek_padj) & lcl_results_xy_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_up[lcl_results_xy_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$MCF7_padj) & lcl_results_xy_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$MCF7_padj) & lcl_results_xy_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$MCF7_padj) & lcl_results_xy_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$MCF7_padj) & lcl_results_xy_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC > 0,])[1],
                                            n=total_lcl_mcf7_exp - dim(lcl_results[! is.na(lcl_results$MCF7_padj) & lcl_results$MCF7_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$C42B_padj) & lcl_results_xy_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$C42B_padj) & lcl_results_xy_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$C42B_padj) & lcl_results_xy_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$C42B_padj) & lcl_results_xy_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC > 0,])[1],
                                            n=total_lcl_c42b_exp - dim(lcl_results[! is.na(lcl_results$C42B_padj) & lcl_results$C42B_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$hek_padj) & lcl_results_xy_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$hek_padj) & lcl_results_xy_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$hek_padj) & lcl_results_xy_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(lcl_results_xy_sig_dn[! is.na(lcl_results_xy_sig_dn$hek_padj) & lcl_results_xy_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC > 0,])[1],
                                            n=total_lcl_hek_exp - dim(lcl_results[! is.na(lcl_results$hek_padj) & lcl_results$hek_log2FC < 0,])[1],
                                            k=dim(lcl_results_xy_sig_dn[lcl_results_xy_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_XY <- myData_XY
myData_plot_XY$Category <- factor(x=myData_plot_XY$Category, levels = c("XY_up","XY_dn"))
myData_plot_XY$Public_data <- factor(x=myData_plot_XY$Public_data, levels = public_data)
myData_plot_XY[myData_plot_XY$Percentage == 0, "Percentage"] <- 0.1

pdf(file="XY_resp_LCL_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_XY, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="LCLs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()

all_data_lcl <- rbind(myData_plot_X, myData_plot_Y, myData_plot_XY)
all_data_lcl$padj <- all_data_lcl$hypergeometric_p * 72
```
## Fibroblasts
```{r}
fib_results <- lcl_fib_mcf7_c42b_hek[! is.na(lcl_fib_mcf7_c42b_hek$fib_padj_X), ]
fib_results_x_sig_up <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_log2FC_X > 0,]
fib_results_x_sig_dn <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_log2FC_X < 0,]
fib_results_y_sig_up <- fib_results[fib_results$fib_padj_Y < 0.05 & fib_results$fib_log2FC_Y > 0,]
fib_results_y_sig_dn <- fib_results[fib_results$fib_padj_Y < 0.05 & fib_results$fib_log2FC_Y < 0,]
fib_results_xy_sig_up <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_log2FC_X > 0 & fib_results$fib_padj_Y < 0.05 & fib_results$fib_log2FC_Y > 0,]
fib_results_xy_sig_dn <- fib_results[fib_results$fib_padj_X < 0.05 & fib_results$fib_log2FC_X < 0 & fib_results$fib_padj_Y < 0.05 & fib_results$fib_log2FC_Y < 0,]


total_fib_mcf7_exp <- dim(fib_results[fib_results$mcf7_expressed == "Yes",])[1]
total_fib_c42b_exp <- dim(fib_results[fib_results$c42b_expressed == "Yes",])[1]
total_fib_hek_exp <- dim(fib_results[fib_results$hek_expressed == "Yes",])[1]

myData <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())

#Up
myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$MCF7_padj) & fib_results_x_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$mcf7_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$MCF7_padj) &
                                                                                            fib_results_x_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$MCF7_padj) & fib_results_x_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$mcf7_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$MCF7_padj) & fib_results_x_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$C42B_padj) & fib_results_x_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$c42b_expressed == "Yes",])[1]) * 100,
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$C42B_padj) & fib_results_x_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$C42B_padj) & fib_results_x_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$C42B_padj) & fib_results_x_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$hek_padj) & fib_results_x_sig_up$hek_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$hek_padj) & fib_results_x_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$hek_padj) & fib_results_x_sig_up$hek_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_up[fib_results_x_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_up[! is.na(fib_results_x_sig_up$hek_padj) & fib_results_x_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_up[fib_results_x_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$MCF7_padj) & fib_results_x_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$MCF7_padj) & fib_results_x_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$MCF7_padj) & fib_results_x_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$MCF7_padj) & fib_results_x_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$C42B_padj) & fib_results_x_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$C42B_padj) & fib_results_x_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$C42B_padj) & fib_results_x_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$C42B_padj) & fib_results_x_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$hek_padj) & fib_results_x_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$hek_padj) & fib_results_x_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData <- rbind(myData, data.frame("Category" = "X_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$hek_padj) & fib_results_x_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(fib_results_x_sig_dn[fib_results_x_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_x_sig_dn[! is.na(fib_results_x_sig_dn$hek_padj) & fib_results_x_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_x_sig_dn[fib_results_x_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_X <- myData
myData_plot_X$Category <- factor(x=myData_plot_X$Category, levels = c("X_up","X_dn"))
myData_plot_X$Public_data <- factor(x=myData_plot_X$Public_data, levels = public_data)
myData_plot_X[myData_plot_X$Percentage == 0, "Percentage"] <- 0.1

pdf(file="X_resp_fib_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_X, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="fibs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()


#Y response
myData_Y <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())
#No change

#Up
myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$MCF7_padj) & fib_results_y_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$MCF7_padj) & fib_results_y_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$MCF7_padj) & fib_results_y_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$MCF7_padj) & fib_results_y_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$C42B_padj) & fib_results_y_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$C42B_padj) & fib_results_y_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$C42B_padj) & fib_results_y_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$C42B_padj) & fib_results_y_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$hek_padj) & fib_results_y_sig_up$hek_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$hek_padj) & fib_results_y_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$hek_padj) & fib_results_y_sig_up$hek_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_up[fib_results_y_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_up[! is.na(fib_results_y_sig_up$hek_padj) & fib_results_y_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_up[fib_results_y_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$MCF7_padj) & fib_results_y_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$MCF7_padj) & fib_results_y_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$MCF7_padj) & fib_results_y_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$MCF7_padj) & fib_results_y_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$C42B_padj) & fib_results_y_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$C42B_padj) & fib_results_y_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$C42B_padj) & fib_results_y_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$C42B_padj) & fib_results_y_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$hek_padj) & fib_results_y_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$hek_padj) & fib_results_y_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_Y <- rbind(myData_Y, data.frame("Category" = "Y_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$hek_padj) & fib_results_y_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(fib_results_y_sig_dn[fib_results_y_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_y_sig_dn[! is.na(fib_results_y_sig_dn$hek_padj) & fib_results_y_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_y_sig_dn[fib_results_y_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_Y <- myData_Y
myData_plot_Y$Category <- factor(x=myData_plot_Y$Category, levels = c("Y_up","Y_dn"))
myData_plot_Y$Public_data <- factor(x=myData_plot_Y$Public_data, levels = public_data)
myData_plot_Y[myData_plot_Y$Percentage == 0, "Percentage"] <- 0.1

pdf(file="Y_resp_fib_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_Y, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="fibs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()


#X and Y response
myData_XY <- data.frame("Category" = character(), "Public_data" = character(), "Percentage" = numeric(), "hypergeometric_p" = numeric())
#No change

#Up
myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$MCF7_padj) & fib_results_xy_sig_up$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$MCF7_padj) & fib_results_xy_sig_up$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$MCF7_padj) & fib_results_xy_sig_up$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$MCF7_padj) & fib_results_xy_sig_up$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$C42B_padj) & fib_results_xy_sig_up$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$C42B_padj) & fib_results_xy_sig_up$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$C42B_padj) & fib_results_xy_sig_up$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$C42B_padj) & fib_results_xy_sig_up$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$hek_padj) & fib_results_xy_sig_up$hek_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$hek_padj) & fib_results_xy_sig_up$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_up","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$hek_padj) & fib_results_xy_sig_up$hek_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_up[fib_results_xy_sig_up$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_up[! is.na(fib_results_xy_sig_up$hek_padj) & fib_results_xy_sig_up$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_up[fib_results_xy_sig_up$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

#Down
myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "mcf7_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$MCF7_padj) & fib_results_xy_sig_dn$MCF7_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$MCF7_padj) & fib_results_xy_sig_dn$MCF7_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "mcf7_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$MCF7_padj) & fib_results_xy_sig_dn$MCF7_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$mcf7_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$MCF7_padj) & fib_results_xy_sig_dn$MCF7_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC > 0,])[1],
                                            n=total_fib_mcf7_exp - dim(fib_results[! is.na(fib_results$MCF7_padj) & fib_results$MCF7_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$mcf7_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "c42b_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$C42B_padj) & fib_results_xy_sig_dn$C42B_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$C42B_padj) & fib_results_xy_sig_dn$C42B_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "c42b_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$C42B_padj) & fib_results_xy_sig_dn$C42B_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$c42b_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$C42B_padj) & fib_results_xy_sig_dn$C42B_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC > 0,])[1],
                                            n=total_fib_c42b_exp - dim(fib_results[! is.na(fib_results$C42B_padj) & fib_results$C42B_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$c42b_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "hek_up", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$hek_padj) & fib_results_xy_sig_dn$hek_log2FC > 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$hek_padj) & fib_results_xy_sig_dn$hek_log2FC > 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_XY <- rbind(myData_XY, data.frame("Category" = "XY_dn","Public_data" = "hek_dn", 
                                   "Percentage" = (dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$hek_padj) & fib_results_xy_sig_dn$hek_log2FC < 0,])[1] /
                                     dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$hek_expressed == "Yes",])[1] * 100),
                                   "hypergeometric_p" = phyper(q=dim(fib_results_xy_sig_dn[! is.na(fib_results_xy_sig_dn$hek_padj) & fib_results_xy_sig_dn$hek_log2FC < 0,])[1] - 1, 
                                            m=dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC > 0,])[1],
                                            n=total_fib_hek_exp - dim(fib_results[! is.na(fib_results$hek_padj) & fib_results$hek_log2FC < 0,])[1],
                                            k=dim(fib_results_xy_sig_dn[fib_results_xy_sig_dn$hek_expressed == "Yes",])[1], lower.tail = FALSE))) 

myData_plot_XY <- myData_XY
myData_plot_XY$Category <- factor(x=myData_plot_XY$Category, levels = c("XY_up","XY_dn"))
myData_plot_XY$Public_data <- factor(x=myData_plot_XY$Public_data, levels = public_data)
myData_plot_XY[myData_plot_XY$Percentage == 0, "Percentage"] <- 0.1

pdf(file="XY_resp_fib_public_ZFX.pdf", height=2, width=3)
ggplot(data=myData_plot_XY, mapping = aes(x=Category, y=Percentage, fill=Public_data)) + 
   geom_bar(position= position_dodge(width=0.8), stat="identity", width=0.7) +
  scale_fill_manual(values = myCols) +
   theme_classic(base_size = 8) + 
  theme(
      axis.text = element_text(color = "black"), 
      axis.ticks = element_line(color = "black", size = 0.25), 
      axis.line = element_line(size = 0.25),
      axis.line.x = element_blank(),
      axis.ticks.x= element_blank(),
      plot.title = element_text(hjust = 0.5),
      axis.text.x = element_text(hjust = 0.5),
      legend.position = "none",
      ) +
  labs(
    title="fibs",
    y="Percent called as direct ZFX targets", x=""
  ) + scale_y_continuous(expand=c(0,0),limits=c(0,70))
dev.off()

all_data_fib <- rbind(myData_plot_X, myData_plot_Y, myData_plot_XY)
all_data_fib$padj <- p.adjust(p=all_data_fib$hypergeometric_p, method="bonferroni")
```
